<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Determining ploidy 1</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-2"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-2');
</script>


<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-1"></script>
<script> 
window.dataLayer = window.dataLayer || []; 
function gtag(){dataLayer.push(arguments)}; 
gtag('js', new Date()); 
gtag('config', 'UA-107144798-1'); 
</script>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">vcfR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="rlanguage.html">R language</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="quick_intro.html">A quick introduction</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="vcfR_object.html">vcfR objects</a>
    </li>
    <li>
      <a href="how_much_memory.html">How much memory</a>
    </li>
    <li>
      <a href="matrices.html">Extracting matrices</a>
    </li>
    <li>
      <a href="tidy_vcfR.html">Tidy vcfR</a>
    </li>
    <li>
      <a href="chromR_object.html">chromR objects</a>
    </li>
    <li>
      <a href="visualization_1.html">Visualization 1</a>
    </li>
    <li>
      <a href="visualization_2.html">Visualization 2</a>
    </li>
    <li>
      <a href="sequence_coverage.html">Sequence coverage</a>
    </li>
    <li>
      <a href="filtering_data.html">Filtering data</a>
    </li>
    <li>
      <a href="ranking_data.html">Ranking data</a>
    </li>
    <li>
      <a href="windowing.html">Windowing</a>
    </li>
    <li>
      <a href="genetic_differentiation.html">Genetic differentiation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    GBS class
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="gbs_class.html">GBS class</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="extract_data.html">Extract data</a>
    </li>
    <li>
      <a href="depth_plot.html">Depth plot</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="censoring_data.html">Censoring data</a>
    </li>
    <li>
      <a href="omitting_data.html">Omitting data</a>
    </li>
    <li>
      <a href="apply.html">Apply</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Ploidy
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ploidy.html">Ploidy</a>
    </li>
    <li>
      <a href="determining_ploidy_1.html">Determining ploidy 1</a>
    </li>
    <li>
      <a href="determining_ploidy_2.html">Determining ploidy 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Export
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="export.html">Overview</a>
    </li>
    <li>
      <a href="export_vcfgz.html">Export to *vcf.gz</a>
    </li>
    <li>
      <a href="export_genind_genclone.html">Genind and Genclone</a>
    </li>
    <li>
      <a href="export_genlight_snpclone.html">Genlight and SNPclone</a>
    </li>
    <li>
      <a href="dnabin.html">DNAbin</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="subset_data_to_1chrom.html">Subset to 1 chromosome</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="vcf_software.html">VCF software</a>
    </li>
    <li>
      <a href="dip_to_hap.html">Haploidizing diploid data</a>
    </li>
    <li>
      <a href="compiling_vcfR.html">Compiling vcfR</a>
    </li>
    <li>
      <a href="reporting_issue.html">Reporting an issue</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<p>
<center>
<h3>vcfR documentation</h3>
by
<br>
Brian J. Knaus and Niklaus J. Gr&uuml;nwald
</center>
</p>




<div id="header">



<h1 class="title toc-ignore">Determining ploidy 1</h1>

</div>


<p>For some organisms ploidy information is unclear. This is important
because current variant callers can need to have ploidy specified at the
time of variant calling (i.e., they cannot infer ploidy). Here I explore
a method to infer ploidy from VCF data.</p>
<div id="input-data" class="section level2">
<h2>Input data</h2>
<p>Data import is performed similar to other examples.</p>
<pre class="r"><code># Load libraries
library(vcfR)
library(pinfsc50)

# Determine file locations
vcf_file &lt;- system.file(&quot;extdata&quot;, &quot;pinf_sc50.vcf.gz&quot;,
                        package = &quot;pinfsc50&quot;)

# Read data into memory
vcf &lt;- read.vcfR(vcf_file, verbose = FALSE)
vcf</code></pre>
<pre><code>## ***** Object of Class vcfR *****
## 18 samples
## 1 CHROMs
## 22,031 variants
## Object size: 22.4 Mb
## 7.929 percent missing data
## *****        *****         *****</code></pre>
</div>
<div id="allele-balance" class="section level2">
<h2>Allele balance</h2>
<p>In high throughput sequencing we sequence each allele multiple times.
For heterozygotes it is assumed that we sequence each allele an equal
number of times. For example, if we sequence a diploid heterozygote at
30X we would expect to sequence each allele 15 times. Some variation
around this expectation may be due to sequencing error. If we sequence a
triploid heterozygote with a genotype of A/A/T at 30X we would expect to
sequence A 20 times and T 10 times. Because we can not determine which A
was sequenced we can only report the sum of all sequences even though we
expect to sequence both A alleles 10 times. For a triploid heterozygote
with a genotype of A/C/T sequenced at 30X would expect to sequence A 10
times, C 10 times and T 10 times.</p>
<pre class="r"><code>knitr::kable(vcf@gt[c(1:2,11,30),1:4])</code></pre>
<table style="width:100%;">
<colgroup>
<col width="14%" />
<col width="28%" />
<col width="28%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">FORMAT</th>
<th align="left">BL2009P4_us23</th>
<th align="left">DDR7602</th>
<th align="left">IN2009T1_us22</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GT:AD:DP:GQ:PL</td>
<td align="left">1|1:0,7:7:21:283,21,0</td>
<td align="left">1|1:0,6:6:18:243,18,0</td>
<td align="left">1|1:0,8:8:24:324,24,0</td>
</tr>
<tr class="even">
<td align="left">GT:AD:DP:GQ:PL</td>
<td align="left">0|0:12,0:12:36:0,36,427</td>
<td align="left">0|0:20,0:20:60:0,60,819</td>
<td align="left">0|0:16,0:16:48:0,48,650</td>
</tr>
<tr class="odd">
<td align="left">GT:AD:DP:GQ:PL</td>
<td align="left">0|1:17,12:29:99:453,0,667</td>
<td align="left">0|0:32,0:32:96:0,96,1433</td>
<td align="left">0|0:40,0:40:99:0,120,1765</td>
</tr>
<tr class="even">
<td align="left">GT:AD:DP:GQ:PL</td>
<td align="left">1|0:7,4:11:99:130,0,260</td>
<td align="left">0|0:16,0:16:48:0,48,677</td>
<td align="left">0|0:26,0:26:78:0,78,1073</td>
</tr>
</tbody>
</table>
<p>The ‘AD’ field in our VCF data includes the depth at which each
allele was sequenced. We can extract this information with the function
<code>extract.gt()</code>.</p>
<pre class="r"><code>ad &lt;- extract.gt(vcf, element = &#39;AD&#39;)</code></pre>
<pre class="r"><code>knitr::kable(ad[c(1:2,11,30),1:4])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">BL2009P4_us23</th>
<th align="left">DDR7602</th>
<th align="left">IN2009T1_us22</th>
<th align="left">LBUS5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Supercontig_1.50_41</td>
<td align="left">0,7</td>
<td align="left">0,6</td>
<td align="left">0,8</td>
<td align="left">0,6</td>
</tr>
<tr class="even">
<td align="left">Supercontig_1.50_136</td>
<td align="left">12,0</td>
<td align="left">20,0</td>
<td align="left">16,0</td>
<td align="left">20,0</td>
</tr>
<tr class="odd">
<td align="left">Supercontig_1.50_616</td>
<td align="left">17,12</td>
<td align="left">32,0</td>
<td align="left">40,0</td>
<td align="left">32,0</td>
</tr>
<tr class="even">
<td align="left">Supercontig_1.50_1887</td>
<td align="left">7,4</td>
<td align="left">16,0</td>
<td align="left">26,0</td>
<td align="left">16,0</td>
</tr>
</tbody>
</table>
<p>Note that these genotypes were called as diploids. This can be
validated by confirming that the genotypes consist of two alleles.
Allele depth appears to be independent of this parameterization. This is
probably because this information can be gathered upstream of the
process of genotype calling. Despite the specification of diploidy, the
allele depth information can include the observation of multiple
alleles.</p>
<pre class="r"><code>#knitr::kable(ad[grep(&quot;[^0],[^0],[^0]&quot;, ad[,&#39;DDR7602&#39;]),1:4])
knitr::kable(ad[grep(&quot;[^0],[^0],[^0]&quot;, ad[,&#39;P13626&#39;])[1:6],c(2:3,12)])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">DDR7602</th>
<th align="left">IN2009T1_us22</th>
<th align="left">P13626</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Supercontig_1.50_105867</td>
<td align="left">0,0,23</td>
<td align="left">12,9,0</td>
<td align="left">8,7,13</td>
</tr>
<tr class="even">
<td align="left">Supercontig_1.50_208264</td>
<td align="left">7,2,0</td>
<td align="left">14,1,0</td>
<td align="left">6,2,4</td>
</tr>
<tr class="odd">
<td align="left">Supercontig_1.50_212192</td>
<td align="left">2,1,0</td>
<td align="left">18,2,0</td>
<td align="left">9,1,6</td>
</tr>
<tr class="even">
<td align="left">Supercontig_1.50_218971</td>
<td align="left">1,0,4</td>
<td align="left">14,3,7</td>
<td align="left">11,2,9</td>
</tr>
<tr class="odd">
<td align="left">Supercontig_1.50_442450</td>
<td align="left">2,3,0</td>
<td align="left">0,7,0</td>
<td align="left">1,5,1</td>
</tr>
<tr class="even">
<td align="left">Supercontig_1.50_485181</td>
<td align="left">29,2,1</td>
<td align="left">42,0,0</td>
<td align="left">29,4,3</td>
</tr>
</tbody>
</table>
<p>This can be seen in the VCF data along with the genotypes as
well.</p>
<pre class="r"><code>#knitr::kable(vcf@gt[grep(&quot;[^0],[^0],[^0]&quot;, ad[,&#39;P13626&#39;]),c(4:5,13)])
knitr::kable(vcf@gt[grep(&quot;[^0],[^0],[^0]&quot;, ad[,&#39;P13626&#39;])[1:6],c(3:4,13)])</code></pre>
<table>
<colgroup>
<col width="33%" />
<col width="34%" />
<col width="32%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">DDR7602</th>
<th align="left">IN2009T1_us22</th>
<th align="left">P13626</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">2|2:0,0,23:23:72:1080,1080,1080,72,72,0</td>
<td align="left">0|1:12,9,0:21:99:308,0,431,351,487,1139</td>
<td align="left">0|2:8,7,13:28:32:644,377,691,0,32,412</td>
</tr>
<tr class="even">
<td align="left">0|1:7,2,0:9:99:105,0,285,126,221,330</td>
<td align="left">0|0:14,1,0:15:0:0,0,585,42,506,529</td>
<td align="left">2|0:6,2,4:12:66:126,66,330,0,124,187</td>
</tr>
<tr class="odd">
<td align="left">0|1:2,1,0:3:36:36,0,70,42,38,74</td>
<td align="left">1|0:18,2,0:20:30:30,0,750,84,574,615</td>
<td align="left">2|0:9,1,6:16:99:149,149,570,0,266,286</td>
</tr>
<tr class="even">
<td align="left">2|0:1,0,4:5:13:90,109,209,0,31,13</td>
<td align="left">0|2:14,3,7:24:86:98,86,493,0,254,344</td>
<td align="left">0|2:11,2,9:22:99:174,186,563,0,210,244</td>
</tr>
<tr class="odd">
<td align="left">1|0:2,3,0:5:99:111,0,113,120,122,241</td>
<td align="left">1|1:0,7,0:7:21:297,21,0,297,21,297</td>
<td align="left">0|1:1,5,1:7:19:147,0,27,127,19,165</td>
</tr>
<tr class="even">
<td align="left">0|0:29,2,1:32:45:0,45,965,45,963,1386</td>
<td align="left">0|0:42,0,0:42:99:0,125,1438,131,1525,1875</td>
<td align="left">0|1:29,4,3:36:20:20,0,874,114,916,1710</td>
</tr>
</tbody>
</table>
<p>The genotypes are reported as diploid genotypes because this was
specified as an argument for the variant caller. Despite this
specification, we can see that more than two alleles can be observed for
a sample at a variant. Here we’ve searched the sample P13626 for teh
presented example.</p>
<p>The function <code>extract.gt()</code> isolates the ‘AD’ data from
the colon delimited string in the VCF data. We expect integer counts for
the number of sequences observed. However, because this data is comma
delimited we need another step before we have integers. We use the
function <code>masplit()</code> to extract the first and second allele.
At that point we have integers so we can use math to create allele
frequencies from the counts.</p>
<pre class="r"><code>allele1 &lt;- masplit(ad, record = 1)
allele2 &lt;- masplit(ad, record = 2)

ad1 &lt;- allele1 / (allele1 + allele2)
ad2 &lt;- allele2 / (allele1 + allele2)</code></pre>
<p>Once we have our allele frequencies we can plot them with a
histogram.</p>
<pre class="r"><code>hist(ad2[,&quot;P17777us22&quot;], breaks = seq(0,1,by=0.02), col = &quot;#1f78b4&quot;, xaxt=&quot;n&quot;)
hist(ad1[,&quot;P17777us22&quot;], breaks = seq(0,1,by=0.02), col = &quot;#a6cee3&quot;, add = TRUE)
axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1), labels=c(0,&quot;1/4&quot;,&quot;1/3&quot;,&quot;1/2&quot;,&quot;2/3&quot;,&quot;3/4&quot;,1))</code></pre>
<p><img src="determining_ploidy_1_files/figure-html/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>The most common class of variant is the homozygote. This overwhelms
the plot. We can remove these so we can focus on the heterozygotes.</p>
<pre class="r"><code>gt &lt;- extract.gt(vcf, element = &#39;GT&#39;)
hets &lt;- is_het(gt)

is.na( ad[ !hets ] ) &lt;- TRUE

allele1 &lt;- masplit(ad, record = 1)
allele2 &lt;- masplit(ad, record = 2)

ad1 &lt;- allele1 / (allele1 + allele2)
ad2 &lt;- allele2 / (allele1 + allele2)

hist(ad2[,&quot;P17777us22&quot;], breaks = seq(0,1,by=0.02), col = &quot;#1f78b4&quot;, xaxt=&quot;n&quot;)
hist(ad1[,&quot;P17777us22&quot;], breaks = seq(0,1,by=0.02), col = &quot;#a6cee3&quot;, add = TRUE)
axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1), labels=c(0,&quot;1/4&quot;,&quot;1/3&quot;,&quot;1/2&quot;,&quot;2/3&quot;,&quot;3/4&quot;,1))</code></pre>
<p><img src="determining_ploidy_1_files/figure-html/unnamed-chunk-9-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>This allows us to focus on the heterozygous variants and we observe a
peak at 0.5 consistent with our expectation. However, the frequencies
range almost completely from 0 to one. This suggests some improvement
could be made.</p>
</div>
<div id="allele-depth" class="section level2">
<h2>Allele Depth</h2>
<p>To get an idea on how to improve our plots we can take a closer look
at allele depth.</p>
<pre class="r"><code>ad &lt;- extract.gt(vcf, element = &#39;AD&#39;)
#ad[1:3,1:4]

allele1 &lt;- masplit(ad, record = 1)
allele2 &lt;- masplit(ad, record = 2)

# Subset to a vector for manipulation.
tmp &lt;- allele1[,&quot;P17777us22&quot;]
#sum(tmp == 0, na.rm = TRUE)
#tmp &lt;- tmp[tmp &gt; 0]
tmp &lt;- tmp[tmp &lt;= 100]

hist(tmp, breaks=seq(0,100,by=1), col=&quot;#808080&quot;, main = &quot;P17777us22&quot;)

sums &lt;- apply(allele1, MARGIN=2, quantile, probs=c(0.15, 0.95), na.rm=TRUE)
sums[,&quot;P17777us22&quot;]</code></pre>
<pre><code>## 15% 95% 
##  19  75</code></pre>
<pre class="r"><code>abline(v=sums[,&quot;P17777us22&quot;], col=2, lwd=2)</code></pre>
<p><img src="determining_ploidy_1_files/figure-html/unnamed-chunk-10-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>Here we have presented the number of times the most abundant allele
was sequenced as a histogram. We see two classes of calls resulting in
peaks at just below 40 and just below 60. These peaks correspond to
heterozygotes and homozygotes respectively. There are a substantial
number of variants with exceptionally high coverage, here we’ve limited
the presentation to just variants below 100. There is also a substantial
peak near zero representing variants sequenced at low coverage.
Filtering this data by imposing some sort of threshold may improve our
other analyses. Here we’ve used the 15% and 95% quantiles.</p>
<p>We can see how this treatment looks on the second most abundant
allele as well.</p>
<pre class="r"><code>tmp &lt;- allele2[,&quot;P17777us22&quot;]
tmp &lt;- tmp[tmp&gt;0]
tmp &lt;- tmp[tmp&lt;=100]

hist(tmp, breaks=seq(1,100,by=1), col=&quot;#808080&quot;, main=&quot;P17777us22&quot;)

sums[,&quot;P17777us22&quot;]</code></pre>
<pre><code>## 15% 95% 
##  19  75</code></pre>
<pre class="r"><code>abline(v=sums[,&quot;P17777us22&quot;], col=2, lwd=2)</code></pre>
<p><img src="determining_ploidy_1_files/figure-html/unnamed-chunk-12-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>We can use this information to filter our data. First we’ll look at
the raw data.</p>
<pre class="r"><code>#vcf &lt;- extract.indels(vcf)
#gq &lt;- extract.gt(vcf, element = &#39;GQ&#39;, as.numeric = TRUE)
#vcf@gt[,-1][ gq &lt; 99 ] &lt;- NA

ad &lt;- extract.gt(vcf, element = &#39;AD&#39;)
allele1 &lt;- masplit(ad, record = 1)
allele2 &lt;- masplit(ad, record = 2)
boxplot(allele1, las=3)</code></pre>
<p><img src="determining_ploidy_1_files/figure-html/unnamed-chunk-13-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#hist(allele1[,&quot;P17777us22&quot;], ylim=c(0,2000),
#     breaks = seq(0,max(allele1[,&quot;P17777us22&quot;], na.rm = TRUE),by=5),
#     xlim=c(0,100))

# Subset to a vector for manipulation.
#tmp &lt;- allele1[,&quot;P17777us22&quot;]
#tmp &lt;- tmp[tmp &lt;= 100]
#hist(tmp, breaks=seq(0,100,by=1), col=&quot;#808080&quot;)</code></pre>
<p>Now we can use our quantiles to censor variants outside of our
desired range by setting them to NA.</p>
<pre class="r"><code>sums &lt;- apply(allele1, MARGIN=2, quantile, probs=c(0.15, 0.95), na.rm=TRUE)
# Allele 1
dp2 &lt;- sweep(allele1, MARGIN=2, FUN = &quot;-&quot;, sums[1,])
#allele1[dp2 &lt; 0] &lt;- NA
vcf@gt[,-1][ dp2 &lt; 0 &amp; !is.na(vcf@gt[,-1]) ] &lt;- NA
dp2 &lt;- sweep(allele1, MARGIN=2, FUN = &quot;-&quot;, sums[2,])
#allele1[dp2 &gt; 0] &lt;- NA
vcf@gt[,-1][dp2 &gt; 0] &lt;- NA
# Allele 2
dp2 &lt;- sweep(allele2, MARGIN=2, FUN = &quot;-&quot;, sums[1,])
vcf@gt[,-1][ dp2 &lt; 0 &amp; !is.na(vcf@gt[,-1]) ] &lt;- NA
dp2 &lt;- sweep(allele2, MARGIN=2, FUN = &quot;-&quot;, sums[2,])
vcf@gt[,-1][dp2 &gt; 0] &lt;- NA

# Hard filter
#dp[dp &lt; 4] &lt;- NA
#vcf@gt[,-1][allele1 &lt; 8] &lt;- NA
#

#adp &lt;- adp[adp&lt;=100]
#adp &lt;- adp[adp&gt;=sums[,&quot;P17777us22&quot;][1]]
#adp &lt;- adp[adp&lt;=sums[,&quot;P17777us22&quot;][2]]

#hist(adp, breaks=seq(0, max(adp, na.rm = TRUE )+1, by=1), col=&quot;#C0C0C0&quot;, xlim = c(0,100))
#axis(side=1,at=1:4*10)
#abline(v=sums[,&quot;P17777us22&quot;], col=2, lwd = 2)

#adp &lt;- allele2[,&quot;P17777us22&quot;]
#adp &lt;- adp[adp&gt;0]
#adp &lt;- adp[adp&lt;=100]
#hist(adp, breaks=seq(0, max(adp, na.rm = TRUE), by=1), col=&quot;#C0C0C0&quot;)
#par(mfrow=c(1,1))
#abline(v=sums[,&quot;P17777us22&quot;], col=2, lwd = 2)</code></pre>
<p>Now we can check out work with another set of box and whisker
plots.</p>
<pre class="r"><code>ad &lt;- extract.gt(vcf, element = &#39;AD&#39;)
allele1 &lt;- masplit(ad, record = 1)
allele2 &lt;- masplit(ad, record = 2)
boxplot(allele1, las=3)</code></pre>
<p><img src="determining_ploidy_1_files/figure-html/unnamed-chunk-15-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#hist(allele1[,&quot;P17777us22&quot;], ylim=c(0,2000), breaks = seq(0,100,by=5) )</code></pre>
<p>Now we can see if our histogram of allele balance has been cleaned
up.</p>
<pre class="r"><code>gt &lt;- extract.gt(vcf, element = &#39;GT&#39;)
hets &lt;- is_het(gt)
is.na( ad[ !hets ] ) &lt;- TRUE

allele1 &lt;- masplit(ad, record = 1)
allele2 &lt;- masplit(ad, record = 2)

ad1 &lt;- allele1 / (allele1 + allele2)
ad2 &lt;- allele2 / (allele1 + allele2)

hist(ad2[,&quot;P17777us22&quot;], breaks = seq(0,1,by=0.02), col = &quot;#1f78b4&quot;, xaxt=&quot;n&quot;, main=&quot;P17777us22&quot;)
hist(ad1[,&quot;P17777us22&quot;], breaks = seq(0,1,by=0.02), col = &quot;#a6cee3&quot;, add = TRUE)
axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1), labels=c(0,&quot;1/4&quot;,&quot;1/3&quot;,&quot;1/2&quot;,&quot;2/3&quot;,&quot;3/4&quot;,1))</code></pre>
<p><img src="determining_ploidy_1_files/figure-html/unnamed-chunk-16-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>We see that the shoulders that were present in the previous plot are
almost entirely removed. This appears to be a dramatic improvement.</p>
<p>While we’ve focused on one sample through most of this example its
important to realize that we’ve been analyzing all of the samples in the
dataset. This means that now that we have one sample that appears to to
have been well processed we can now examine the other samples as
well.</p>
<pre class="r"><code>hist(ad2[,&quot;P13626&quot;], breaks = seq(0,1,by=0.02), col = &quot;#33a02c&quot;, xaxt=&quot;n&quot;, main=&quot;P13626&quot;)
hist(ad1[,&quot;P13626&quot;], breaks = seq(0,1,by=0.02), col = &quot;#b2df8a&quot;, add = TRUE)
axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1), labels=c(0,&quot;1/4&quot;,&quot;1/3&quot;,&quot;1/2&quot;,&quot;2/3&quot;,&quot;3/4&quot;,1))</code></pre>
<p><img src="determining_ploidy_1_files/figure-html/unnamed-chunk-17-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>This is an example of a sample that appears triploid. The greatest
density of the allele frequencies are at 1/3 and 2/3 as we would expect
from a triploid. This demonstrates how these methods can be used to
infer ploidy among diploids, triploids and possibly higher ploids.</p>
</div>

<center>
<hr class="style1">
<p>Copyright &copy; 2017, 2018 Brian J. Knaus. All rights reserved.</p>
<p>USDA Agricultural Research Service, Horticultural Crops Research Lab.</p>
</center>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
