<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Determining ploidy 2</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-2"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-2');
</script>


<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-1"></script>
<script> 
window.dataLayer = window.dataLayer || []; 
function gtag(){dataLayer.push(arguments)}; 
gtag('js', new Date()); 
gtag('config', 'UA-107144798-1'); 
</script>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">vcfR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="rlanguage.html">R language</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="quick_intro.html">A quick introduction</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="vcfR_object.html">vcfR objects</a>
    </li>
    <li>
      <a href="how_much_memory.html">How much memory</a>
    </li>
    <li>
      <a href="matrices.html">Extracting matrices</a>
    </li>
    <li>
      <a href="tidy_vcfR.html">Tidy vcfR</a>
    </li>
    <li>
      <a href="chromR_object.html">chromR objects</a>
    </li>
    <li>
      <a href="visualization_1.html">Visualization 1</a>
    </li>
    <li>
      <a href="visualization_2.html">Visualization 2</a>
    </li>
    <li>
      <a href="sequence_coverage.html">Sequence coverage</a>
    </li>
    <li>
      <a href="filtering_data.html">Filtering data</a>
    </li>
    <li>
      <a href="ranking_data.html">Ranking data</a>
    </li>
    <li>
      <a href="windowing.html">Windowing</a>
    </li>
    <li>
      <a href="genetic_differentiation.html">Genetic differentiation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    GBS class
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="gbs_class.html">GBS class</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="extract_data.html">Extract data</a>
    </li>
    <li>
      <a href="depth_plot.html">Depth plot</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="censoring_data.html">Censoring data</a>
    </li>
    <li>
      <a href="omitting_data.html">Omitting data</a>
    </li>
    <li>
      <a href="apply.html">Apply</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Ploidy
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ploidy.html">Ploidy</a>
    </li>
    <li>
      <a href="determining_ploidy_1.html">Determining ploidy 1</a>
    </li>
    <li>
      <a href="determining_ploidy_2.html">Determining ploidy 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Export
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="export.html">Overview</a>
    </li>
    <li>
      <a href="export_vcfgz.html">Export to *vcf.gz</a>
    </li>
    <li>
      <a href="export_genind_genclone.html">Genind and Genclone</a>
    </li>
    <li>
      <a href="export_genlight_snpclone.html">Genlight and SNPclone</a>
    </li>
    <li>
      <a href="dnabin.html">DNAbin</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="subset_data_to_1chrom.html">Subset to 1 chromosome</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="vcf_software.html">VCF software</a>
    </li>
    <li>
      <a href="dip_to_hap.html">Haploidizing diploid data</a>
    </li>
    <li>
      <a href="compiling_vcfR.html">Compiling vcfR</a>
    </li>
    <li>
      <a href="reporting_issue.html">Reporting an issue</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<p>
<center>
<h3>vcfR documentation</h3>
by
<br>
Brian J. Knaus and Niklaus J. Gr&uuml;nwald
</center>
</p>




<div id="header">



<h1 class="title toc-ignore">Determining ploidy 2</h1>

</div>


<p>Previously we showed how to create histograms based on the ratio of
alleles observed at heterozygous positions. These histograms may present
desireable perspectives when a constant ploidy is expected.
Investigators interested in whether there is variation in copy number
along a chromosome may require a more detailed perspective. Here we
present a perspective of copy number variation along the length of a
chromosome.</p>
<p>Instead of using all of the data each chromosome can be divided into
windows of user specified width. For each of these windows a numerical
summary of where the peak would be is made. Creating a numerical summary
is important because when we move to genome scale data it is easy to
generate more windows than we can manually curate. We can visualize some
of this data, but the numerical summary will allow us to process many
windows over many samples.</p>
<div id="input-data" class="section level2">
<h2>Input data</h2>
<p>Data import is performed similar to other examples.</p>
<pre class="r"><code># Load libraries
library(vcfR)
library(pinfsc50)

# Determine file locations
vcf_file &lt;- system.file(&quot;extdata&quot;, &quot;pinf_sc50.vcf.gz&quot;,
                        package = &quot;pinfsc50&quot;)

# Read data into memory
vcf &lt;- read.vcfR(vcf_file, verbose = FALSE)
vcf</code></pre>
<pre><code>## ***** Object of Class vcfR *****
## 18 samples
## 1 CHROMs
## 22,031 variants
## Object size: 22.4 Mb
## 7.929 percent missing data
## *****        *****         *****</code></pre>
</div>
<div id="depth-filtering" class="section level2">
<h2>Depth filtering</h2>
<p>We’ll also filter on depth by removng high and low coverage variants.
The thresholds for this filtering will be determined by quantiles.
Researchers may want to explore other methods available in R (e.g., fit
a distribution, mixture models, etc.).</p>
<p>First we extract the alleles and create frequencies</p>
<pre class="r"><code>ad &lt;- extract.gt(vcf, element = &#39;AD&#39;)

allele1 &lt;- masplit(ad, record = 1)
allele2 &lt;- masplit(ad, record = 2)

ad1 &lt;- allele1 / (allele1 + allele2)
ad2 &lt;- allele2 / (allele1 + allele2)</code></pre>
<p>We create thresholds based on the most abundant allele as
follows.</p>
<pre class="r"><code>dp &lt;- allele1
#sums &lt;- apply(dp, MARGIN=2, quantile, probs=c(0.15, 0.95), na.rm=TRUE)
sums &lt;- apply(dp, MARGIN=2, quantile, probs=c(0.1, 0.9), na.rm=TRUE)

par(mfrow=c(4,3))
par(mar=c(2,2,1,1))
par(oma=c(1,1,0,0))

for(i in 1:12){
  hist(allele1[,i], breaks = seq(0,1e3,by=1), xlim=c(0,100), col=8, main=&quot;&quot;, xlab=&quot;&quot;, ylab=&quot;&quot;)
  title(main = colnames(allele1)[i])
  abline(v=sums[,i], col=2)
}
title(xlab = &quot;Depth&quot;, line=0, outer = TRUE, font=2)
title(ylab = &quot;Count&quot;, line=0, outer = TRUE, font=2)</code></pre>
<p><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-3-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mar=c(5,4,4,2))
par(oma=c(0,0,0,0))</code></pre>
<p>We performa similar operation for the second most abundant
allele.</p>
<pre class="r"><code>par(mfrow=c(4,3))
par(mar=c(2,2,1,1))
par(oma=c(1,1,0,0))

for(i in 1:12){
  tmp &lt;- allele2[,i]
  tmp &lt;- tmp[ tmp &gt; 0 ]
  hist(tmp, breaks = seq(0,1e3,by=1), xlim=c(0,100), 
#       ylim=c(0,1000),
       col=8, main=&quot;&quot;, xlab=&quot;&quot;, ylab=&quot;&quot;)
  title(main = colnames(allele1)[i])
}
title(xlab = &quot;Depth&quot;, line=0, outer = TRUE, font=2)
title(ylab = &quot;Count&quot;, line=0, outer = TRUE, font=2)</code></pre>
<p><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow=c(1,1))
par(mar=c(5,4,4,2))
par(oma=c(0,0,0,0))</code></pre>
</div>
<div id="find-peaks-of-density" class="section level2">
<h2>Find peaks of density</h2>
<p>In order to avoid assumptions about the distribution of the allele
balance ratios we employ a non-parametric method. The data are binned
into bins of user specified widths and the bin with the highest density
is chosen as the frequency. This is our numerical summary of our peak
density and will be retained in a matrix of samples and windows.</p>
<pre class="r"><code># Filter on depth quantiles.
sums &lt;- apply(allele1, MARGIN=2, quantile, probs=c(0.1, 0.9), na.rm=TRUE)
# Allele 1
dp2 &lt;- sweep(allele1, MARGIN=2, FUN = &quot;-&quot;, sums[1,])
#allele1[dp2 &lt; 0] &lt;- NA
vcf@gt[,-1][ dp2 &lt; 0 &amp; !is.na(vcf@gt[,-1]) ] &lt;- NA
dp2 &lt;- sweep(allele1, MARGIN=2, FUN = &quot;-&quot;, sums[2,])
#allele1[dp2 &gt; 0] &lt;- NA
vcf@gt[,-1][dp2 &gt; 0] &lt;- NA
# Allele 2
dp2 &lt;- sweep(allele2, MARGIN=2, FUN = &quot;-&quot;, sums[1,])
vcf@gt[,-1][ dp2 &lt; 0 &amp; !is.na(vcf@gt[,-1]) ] &lt;- NA
dp2 &lt;- sweep(allele2, MARGIN=2, FUN = &quot;-&quot;, sums[2,])
vcf@gt[,-1][dp2 &gt; 0] &lt;- NA

# Censor homozygotes.
gt &lt;- extract.gt(vcf, element = &#39;GT&#39;)
hets &lt;- is_het(gt)
is.na( vcf@gt[,-1][ !hets ] ) &lt;- TRUE




# Extract allele depths
ad &lt;- extract.gt(vcf, element = &#39;AD&#39;)
allele1 &lt;- masplit(ad, record = 1)
allele2 &lt;- masplit(ad, record = 2)
ad1 &lt;- allele1 / (allele1 + allele2)
ad2 &lt;- allele2 / (allele1 + allele2)

# Parameters
#winsize &lt;- 1e5
#
winsize &lt;- 2e5
#bin_width &lt;- 0.1
#bin_width &lt;- 0.05
#bin_width &lt;- 0.025
#
bin_width &lt;- 0.02
#bin_width &lt;- 0.01


# Find peaks
freq1 &lt;- ad1/(ad1+ad2)
freq2 &lt;- ad2/(ad1+ad2)
myPeaks1 &lt;- freq_peak(freq1, getPOS(vcf), winsize = winsize, bin_width = bin_width)
#myCounts1 &lt;- freq_peak(freq1, getPOS(vcf), winsize = winsize, bin_width = bin_width, count = TRUE)
is.na(myPeaks1$peaks[myPeaks1$counts &lt; 20]) &lt;- TRUE
myPeaks2 &lt;- freq_peak(freq2, getPOS(vcf), winsize = winsize, bin_width = bin_width, lhs = FALSE)
#myCounts2 &lt;- freq_peak(freq2, getPOS(vcf), winsize = winsize, bin_width = bin_width, count = TRUE)
is.na(myPeaks2$peaks[myPeaks2$counts &lt; 20]) &lt;- TRUE</code></pre>
<p>As a proof of concep we can visualize plots of the major allele with
a red horizontal line at our predicted peak. We repeat this for the
minor allele.</p>
<pre class="r"><code>par(mfrow=c(4,4))
par(mar=c(2,2,1,1))
par(oma=c(1,1,0,0))

mySample &lt;- &quot;BL2009P4_us23&quot;
for(i in 1:4){
  hist(freq1[ myPeaks1$wins[i,&#39;START_row&#39;]:myPeaks1$wins[i,&#39;END_row&#39;], mySample ],
       breaks = seq(0,1,by=bin_width), xlim=c(0,1), col=8, main = &quot;&quot;, xaxt=&#39;n&#39;)
  axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1),
     labels=c(0,&#39;1/4&#39;,&#39;1/3&#39;,&#39;1/2&#39;,&#39;2/3&#39;,&#39;3/4&#39;,1), las=1)
  abline(v=myPeaks1$peaks[i,mySample], col=2)
  if(i==2){ title(main=mySample) }
}

mySample &lt;- &quot;DDR7602&quot;
for(i in 1:4){
  hist(freq1[ myPeaks1$wins[i,&#39;START_row&#39;]:myPeaks1$wins[i,&#39;END_row&#39;], mySample ], 
       breaks = seq(0,1,by=bin_width), xlim=c(0,1), col=8, main = &quot;&quot;, xaxt=&#39;n&#39;)
  axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1),
     labels=c(0,&#39;1/4&#39;,&#39;1/3&#39;,&#39;1/2&#39;,&#39;2/3&#39;,&#39;3/4&#39;,1), las=1)
  abline(v=myPeaks1$peaks[i,mySample], col=2)
  if(i==2){ title(main=mySample) }
}

mySample &lt;- &quot;IN2009T1_us22&quot;
for(i in 1:4){
  hist(freq1[ myPeaks1$wins[i,&#39;START_row&#39;]:myPeaks1$wins[i,&#39;END_row&#39;], mySample ],
       breaks = seq(0,1,by=bin_width), xlim=c(0,1), col=8, main = &quot;&quot;, xaxt=&#39;n&#39;)
  axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1),
     labels=c(0,&#39;1/4&#39;,&#39;1/3&#39;,&#39;1/2&#39;,&#39;2/3&#39;,&#39;3/4&#39;,1), las=1)
  abline(v=myPeaks1$peaks[i,mySample], col=2)
  if(i==2){ title(main=mySample) }
}

mySample &lt;- &quot;P17777us22&quot;
for(i in 1:4){
  hist(freq1[ myPeaks1$wins[i,&#39;START_row&#39;]:myPeaks1$wins[i,&#39;END_row&#39;], mySample ],
       breaks = seq(0,1,by=bin_width), xlim=c(0,1), col=8, main = &quot;&quot;, xaxt=&#39;n&#39;)
  axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1),
     labels=c(0,&#39;1/4&#39;,&#39;1/3&#39;,&#39;1/2&#39;,&#39;2/3&#39;,&#39;3/4&#39;,1), las=1)
  abline(v=myPeaks1$peaks[i,mySample], col=2)
  if(i==2){ title(main=mySample) }
}</code></pre>
<p><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-5-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow=c(1,1))
par(mar=c(5,4,4,2))
par(oma=c(0,0,0,0))</code></pre>
<p>Second most abundant allele.</p>
<pre class="r"><code>par(mfrow=c(4,4))
par(mar=c(2,2,1,1))
par(oma=c(1,1,0,0))

mySample &lt;- &quot;BL2009P4_us23&quot;
for(i in 1:4){
  hist(freq2[ myPeaks1$wins[i,&#39;START_row&#39;]:myPeaks1$wins[i,&#39;END_row&#39;], mySample ],
       breaks = seq(0,1,by=bin_width), xlim=c(0,1), col=8, main = &quot;&quot;, xaxt=&#39;n&#39;)
  axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1),
     labels=c(0,&#39;1/4&#39;,&#39;1/3&#39;,&#39;1/2&#39;,&#39;2/3&#39;,&#39;3/4&#39;,1), las=1)
  abline(v=myPeaks2$peaks[i,mySample], col=2)
  if(i==2){ title(main=mySample) }
}

mySample &lt;- &quot;DDR7602&quot;
for(i in 1:4){
  hist(freq2[ myPeaks1$wins[i,&#39;START_row&#39;]:myPeaks1$wins[i,&#39;END_row&#39;], mySample ],
       breaks = seq(0,1,by=bin_width), xlim=c(0,1), col=8, main = &quot;&quot;, xaxt=&#39;n&#39;)
  axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1),
     labels=c(0,&#39;1/4&#39;,&#39;1/3&#39;,&#39;1/2&#39;,&#39;2/3&#39;,&#39;3/4&#39;,1), las=1)
  abline(v=myPeaks2$peaks[i,mySample], col=2)
  if(i==2){ title(main=mySample) }
}

mySample &lt;- &quot;IN2009T1_us22&quot;
for(i in 1:4){
  hist(freq2[ myPeaks1$wins[i,&#39;START_row&#39;]:myPeaks1$wins[i,&#39;END_row&#39;], mySample ],
       breaks = seq(0,1,by=bin_width), xlim=c(0,1), col=8, main = &quot;&quot;, xaxt=&#39;n&#39;)
  axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1),
     labels=c(0,&#39;1/4&#39;,&#39;1/3&#39;,&#39;1/2&#39;,&#39;2/3&#39;,&#39;3/4&#39;,1), las=1)
  abline(v=myPeaks2$peaks[i,mySample], col=2)
  if(i==2){ title(main=mySample) }
}

mySample &lt;- &quot;P17777us22&quot;
for(i in 1:4){
  hist(freq2[ myPeaks2$wins[i,&#39;START_row&#39;]:myPeaks1$wins[i,&#39;END_row&#39;], mySample ],
       breaks = seq(0,1,by=bin_width), xlim=c(0,1), col=8, main = &quot;&quot;, xaxt=&#39;n&#39;)
  axis(side=1, at=c(0,0.25,0.333,0.5,0.666,0.75,1),
     labels=c(0,&#39;1/4&#39;,&#39;1/3&#39;,&#39;1/2&#39;,&#39;2/3&#39;,&#39;3/4&#39;,1), las=1)
  abline(v=myPeaks2$peaks[i,mySample], col=2)
  if(i==2){ title(main=mySample) }
}</code></pre>
<p><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-6-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow=c(1,1))
par(mar=c(5,4,4,2))
par(oma=c(0,0,0,0))</code></pre>
</div>
<div id="visualization" class="section level2">
<h2>Visualization</h2>
<p>There are a number of ways to visualize this data. One way is to turn
our histogram on its side, as a form of marginal summary, and plot our
window summaries.</p>
<pre class="r"><code>i &lt;- 2

layout(matrix(1:2, nrow=1), widths = c(4,1))
par(mar=c(5,4,4,0))

mySample &lt;- colnames(freq1)[i]
plot(getPOS(vcf), freq1[,mySample], ylim=c(0,1), type=&quot;n&quot;, yaxt=&#39;n&#39;, 
     main = mySample, xlab = &quot;POS&quot;, ylab = &quot;Allele balance&quot;)
axis(side=2, at=c(0,0.25,0.333,0.5,0.666,0.75,1), 
     labels=c(0,&#39;1/4&#39;,&#39;1/3&#39;,&#39;1/2&#39;,&#39;2/3&#39;,&#39;3/4&#39;,1), las=1)
abline(h=c(0.25,0.333,0.5,0.666,0.75), col=8)
points(getPOS(vcf), freq1[,mySample], pch = 20, col= &quot;#A6CEE344&quot;)
points(getPOS(vcf), freq2[,mySample], pch = 20, col= &quot;#1F78B444&quot;)
segments(x0=myPeaks1$wins[,&#39;START_pos&#39;], y0=myPeaks1$peaks[,mySample],
         x1=myPeaks1$wins[,&#39;END_pos&#39;], lwd=3)
segments(x0=myPeaks1$wins[,&#39;START_pos&#39;], y0=myPeaks2$peaks[,mySample],
         x1=myPeaks1$wins[,&#39;END_pos&#39;], lwd=3)

bp1 &lt;- hist(freq1[,mySample], breaks = seq(0,1,by=bin_width), plot = FALSE)
bp2 &lt;- hist(freq2[,mySample], breaks = seq(0,1,by=bin_width), plot = FALSE)

par(mar=c(5,1,4,2))
barplot(height=bp1$counts, width=0.02,  space = 0, horiz = T, add = FALSE, col=&quot;#A6CEE3&quot;)
barplot(height=bp2$counts, width=0.02,  space = 0, horiz = T, add = TRUE, col=&quot;#1F78B4&quot;)</code></pre>
<p><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-7-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mar=c(5,4,4,2))
par(mfrow=c(1,1))</code></pre>
<p>Once we determine how to make one plot we can loop over all the
samples.</p>
<p><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-2.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-3.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-4.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-5.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-6.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-7.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-8.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-9.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-10.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-11.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-12.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-13.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-14.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-15.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-16.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-17.png" width="768" style="display: block; margin: auto;" /><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-8-18.png" width="768" style="display: block; margin: auto;" /></p>
<p>We can also create a matrix of plots.</p>
<pre class="r"><code>par(mfrow=c(4,3))
par(mar=c(2,2,1,1))
par(oma=c(1,1,0,0))


for(i in 1:12){
mySample &lt;- colnames(freq1)[i]
plot(getPOS(vcf), freq1[,mySample], ylim=c(0,1), type=&quot;n&quot;, yaxt=&#39;n&#39;, 
     main = mySample, xlab = &quot;POS&quot;, ylab = &quot;Allele balance&quot;)
axis(side=2, at=c(0,0.25,0.333,0.5,0.666,0.75,1), 
     labels=c(0,&#39;1/4&#39;,&#39;1/3&#39;,&#39;1/2&#39;,&#39;2/3&#39;,&#39;3/4&#39;,1), las=1)
abline(h=c(0.25,0.333,0.5,0.666,0.75), col=8)
points(getPOS(vcf), freq1[,mySample], pch = 20, col= &quot;#A6CEE344&quot;)
points(getPOS(vcf), freq2[,mySample], pch = 20, col= &quot;#1F78B444&quot;)
segments(x0=myPeaks1$wins[,&#39;START_pos&#39;], y0=myPeaks1$peaks[,mySample],
         x1=myPeaks1$wins[,&#39;END_pos&#39;], lwd=3)
segments(x0=myPeaks1$wins[,&#39;START_pos&#39;], y0=myPeaks2$peaks[,mySample],
         x1=myPeaks1$wins[,&#39;END_pos&#39;], lwd=3)
}</code></pre>
<p><img src="determining_ploidy_2_files/figure-html/unnamed-chunk-9-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow=c(1,1))
par(mar=c(5,4,4,2))
par(oma=c(0,0,0,0))</code></pre>
<p>We now have a way to create windows of allele balance ratios. We also
have several ways to visualize this data. Perhaps most importantly, we
have a numerical summary of the peak in each window so we can loop over
all of the chromosomes in our reference.</p>
</div>

<center>
<hr class="style1">
<p>Copyright &copy; 2017, 2018 Brian J. Knaus. All rights reserved.</p>
<p>USDA Agricultural Research Service, Horticultural Crops Research Lab.</p>
</center>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
