<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Genetic differentiation</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-2"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-2');
</script>


<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-1"></script>
<script> 
window.dataLayer = window.dataLayer || []; 
function gtag(){dataLayer.push(arguments)}; 
gtag('js', new Date()); 
gtag('config', 'UA-107144798-1'); 
</script>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">vcfR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="rlanguage.html">R language</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="quick_intro.html">A quick introduction</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="vcfR_object.html">vcfR objects</a>
    </li>
    <li>
      <a href="how_much_memory.html">How much memory</a>
    </li>
    <li>
      <a href="matrices.html">Extracting matrices</a>
    </li>
    <li>
      <a href="tidy_vcfR.html">Tidy vcfR</a>
    </li>
    <li>
      <a href="chromR_object.html">chromR objects</a>
    </li>
    <li>
      <a href="visualization_1.html">Visualization 1</a>
    </li>
    <li>
      <a href="visualization_2.html">Visualization 2</a>
    </li>
    <li>
      <a href="sequence_coverage.html">Sequence coverage</a>
    </li>
    <li>
      <a href="filtering_data.html">Filtering data</a>
    </li>
    <li>
      <a href="ranking_data.html">Ranking data</a>
    </li>
    <li>
      <a href="windowing.html">Windowing</a>
    </li>
    <li>
      <a href="genetic_differentiation.html">Genetic differentiation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    GBS class
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="gbs_class.html">GBS class</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="extract_data.html">Extract data</a>
    </li>
    <li>
      <a href="depth_plot.html">Depth plot</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="censoring_data.html">Censoring data</a>
    </li>
    <li>
      <a href="omitting_data.html">Omitting data</a>
    </li>
    <li>
      <a href="apply.html">Apply</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Ploidy
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ploidy.html">Ploidy</a>
    </li>
    <li>
      <a href="determining_ploidy_1.html">Determining ploidy 1</a>
    </li>
    <li>
      <a href="determining_ploidy_2.html">Determining ploidy 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Export
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="export.html">Overview</a>
    </li>
    <li>
      <a href="export_vcfgz.html">Export to *vcf.gz</a>
    </li>
    <li>
      <a href="export_genind_genclone.html">Genind and Genclone</a>
    </li>
    <li>
      <a href="export_genlight_snpclone.html">Genlight and SNPclone</a>
    </li>
    <li>
      <a href="dnabin.html">DNAbin</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="subset_data_to_1chrom.html">Subset to 1 chromosome</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="vcf_software.html">VCF software</a>
    </li>
    <li>
      <a href="dip_to_hap.html">Haploidizing diploid data</a>
    </li>
    <li>
      <a href="compiling_vcfR.html">Compiling vcfR</a>
    </li>
    <li>
      <a href="reporting_issue.html">Reporting an issue</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<p>
<center>
<h3>vcfR documentation</h3>
by
<br>
Brian J. Knaus and Niklaus J. Gr&uuml;nwald
</center>
</p>




<div id="header">



<h1 class="title toc-ignore">Genetic differentiation</h1>

</div>


<p>A fundamental question to most population studies is whether
populations are diverse and whether this diversity is shared among the
populations? To address the question of within population diversity
geneticists typically report heterozygosity. This is the probability
that two alleles randomly chosen from a population will be different.
Ecologists may know this as Simpson’s Index <span
class="citation">(Simpson 1949)</span>. To address differentiation
population geneticists typically utilize <span
class="math inline">\(F_{ST}\)</span> or one of its analogues.
Population differentiation measured by <span
class="math inline">\(F_{ST}\)</span> was originally proposed by Sewall
Wright <span class="citation">(Wright 1949)</span>. This was later
extended to a method based of diversity by Masatoshi Nei <span
class="citation">(Nei 1973)</span>. As researchers applied these metrics
to microsatellites, genetic markers with a large number of alleles, it
became clear that Nei’s measure would not correctly range from zero to
one, so Philip Hedrick proposed a correction <span
class="citation">(Hedrick 2005)</span>. More recently, Lou Jost proposed
another alternative <span class="citation">(Jost 2008)</span>. You can
tell a topic is popular when so many variants of it are generated. And
there are more variants than I’ve mentioned here. A nice discussion as
to which measure may be appropriate for your data was posted to the
Molecular Ecologist blog titled <a
href="http://www.molecularecologist.com/2011/03/should-i-use-fst-gst-or-d-2/">should
I use <span class="math inline">\(F_{ST}\)</span>, <span
class="math inline">\(G_{ST}\)</span> or <span
class="math inline">\(D\)</span>?</a>.</p>
<p>In <code>vcfR</code> I’ve implemented the function
<code>genetic_diff()</code> to measure population diversity and
differentiation. Because VCF data typically do not include population
information we’ll have to supply it as a factor. The method ‘nei’
employed here is based on the methods reported by Hedrick <span
class="citation">(Hedrick 2005)</span>. The exception is that the
heterozygosities are weighted by the number of alleles observed in each
population. This was inspired by <code>hierfstat::pairwise.fst()</code>
which uses the number of individuals observed in each population to
weight the heterozygosities. By using the number of alleles observed
instead of the number of individuals we remove an assumption about how
many alleles each individual may contribute. That is, we should be able
to accomodate samples of mixed ploidy.</p>
<pre class="r"><code>library(vcfR)
data(vcfR_example)
pop &lt;- as.factor(c(&quot;us&quot;, &quot;eu&quot;, &quot;us&quot;, &quot;af&quot;, &quot;eu&quot;, &quot;us&quot;, &quot;mx&quot;, &quot;eu&quot;, &quot;eu&quot;, &quot;sa&quot;, &quot;mx&quot;, &quot;sa&quot;, &quot;us&quot;, &quot;sa&quot;, &quot;Pmir&quot;, &quot;us&quot;, &quot;eu&quot;, &quot;eu&quot;))
myDiff &lt;- genetic_diff(vcf, pops = pop, method = &#39;nei&#39;)
knitr::kable(head(myDiff[,1:15]))</code></pre>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="3%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="7%" />
<col width="5%" />
<col width="5%" />
<col width="9%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
<col width="6%" />
<col width="4%" />
<col width="4%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">CHROM</th>
<th align="left">POS</th>
<th align="right">Hs_af</th>
<th align="right">Hs_eu</th>
<th align="right">Hs_mx</th>
<th align="right">Hs_Pmir</th>
<th align="right">Hs_sa</th>
<th align="right">Hs_us</th>
<th align="right">Ht</th>
<th align="right">n_af</th>
<th align="right">n_eu</th>
<th align="right">n_mx</th>
<th align="right">n_Pmir</th>
<th align="right">n_sa</th>
<th align="right">n_us</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Supercontig_1.50</td>
<td align="left">2</td>
<td align="right">0</td>
<td align="right">0.0</td>
<td align="right">0.000</td>
<td align="right">0.5</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.0798611</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td align="left">Supercontig_1.50</td>
<td align="left">246</td>
<td align="right">NaN</td>
<td align="right">0.0</td>
<td align="right">0.375</td>
<td align="right">NaN</td>
<td align="right">0.000</td>
<td align="right">0.50</td>
<td align="right">0.3512397</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">Supercontig_1.50</td>
<td align="left">549</td>
<td align="right">NaN</td>
<td align="right">0.0</td>
<td align="right">NaN</td>
<td align="right">NaN</td>
<td align="right">NaN</td>
<td align="right">0.50</td>
<td align="right">0.4444444</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">Supercontig_1.50</td>
<td align="left">668</td>
<td align="right">NaN</td>
<td align="right">0.5</td>
<td align="right">0.000</td>
<td align="right">NaN</td>
<td align="right">0.000</td>
<td align="right">0.50</td>
<td align="right">0.5000000</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">Supercontig_1.50</td>
<td align="left">765</td>
<td align="right">0</td>
<td align="right">0.0</td>
<td align="right">0.000</td>
<td align="right">0.0</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.1107266</td>
<td align="right">2</td>
<td align="right">12</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">10</td>
</tr>
<tr class="even">
<td align="left">Supercontig_1.50</td>
<td align="left">780</td>
<td align="right">0</td>
<td align="right">0.0</td>
<td align="right">0.000</td>
<td align="right">0.0</td>
<td align="right">0.375</td>
<td align="right">0.18</td>
<td align="right">0.1244444</td>
<td align="right">2</td>
<td align="right">8</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">10</td>
</tr>
</tbody>
</table>
<p>The function returns the chromosome and position of each variant as
provided in the VCF data. This should allow you to align its output with
the VCF data. The heterozygosities for each population are reported as
well as the total heterozygosity, followed by the number of alleles
observed in each population. Note that in some populations zero alleles
were observed. Populations with zero alleles reported heterozygosities
of ‘NaN’ because of this absence of data.</p>
<pre class="r"><code>knitr::kable(head(myDiff[,16:19]))</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">Gst</th>
<th align="right">Htmax</th>
<th align="right">Gstmax</th>
<th align="right">Gprimest</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0.4782609</td>
<td align="right">0.7951389</td>
<td align="right">0.9475983</td>
<td align="right">0.5047085</td>
</tr>
<tr class="even">
<td align="right">NaN</td>
<td align="right">0.8057851</td>
<td align="right">NaN</td>
<td align="right">NaN</td>
</tr>
<tr class="odd">
<td align="right">NaN</td>
<td align="right">0.6666667</td>
<td align="right">NaN</td>
<td align="right">NaN</td>
</tr>
<tr class="even">
<td align="right">NaN</td>
<td align="right">0.8125000</td>
<td align="right">NaN</td>
<td align="right">NaN</td>
</tr>
<tr class="odd">
<td align="right">1.0000000</td>
<td align="right">0.7543253</td>
<td align="right">1.0000000</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="right">0.1160714</td>
<td align="right">0.8000000</td>
<td align="right">0.8625000</td>
<td align="right">0.1345756</td>
</tr>
</tbody>
</table>
<p>The remaining columns contain <span
class="math inline">\(G_{ST}\)</span>, the maximum heterozygosity, the
maximum <span class="math inline">\(G_{ST}\)</span> and finally <span
class="math inline">\(G&#39;_{ST}\)</span>. The maximum heterozygosity
and the maximum <span class="math inline">\(G_{ST}\)</span> are
intermediary values used to calculate <span
class="math inline">\(G&#39;_{ST}\)</span>. They are typically not
reported but provide values to help validate that <span
class="math inline">\(G&#39;_{ST}\)</span> was calculated correctly.
Note that the populations that had zero alleles and therefore a
heterozygosity of ‘NaN’ contributed to <span
class="math inline">\(G_{ST}\)</span>s that were also ‘NaN’. To avoid
this you may want to consider omitting populations with a small sample
size or that contain a large amount of missing data.</p>
<p>We now have information for each variant in the VCF data. Because
this is typically a large quantity of information, we’ll want to
summarize it. One way is to take averages of the data.</p>
<pre class="r"><code>knitr::kable(round(colMeans(myDiff[,c(3:9,16,19)], na.rm = TRUE), digits = 3))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Hs_af</td>
<td align="right">0.176</td>
</tr>
<tr class="even">
<td align="left">Hs_eu</td>
<td align="right">0.188</td>
</tr>
<tr class="odd">
<td align="left">Hs_mx</td>
<td align="right">0.168</td>
</tr>
<tr class="even">
<td align="left">Hs_Pmir</td>
<td align="right">0.052</td>
</tr>
<tr class="odd">
<td align="left">Hs_sa</td>
<td align="right">0.198</td>
</tr>
<tr class="even">
<td align="left">Hs_us</td>
<td align="right">0.155</td>
</tr>
<tr class="odd">
<td align="left">Ht</td>
<td align="right">0.247</td>
</tr>
<tr class="even">
<td align="left">Gst</td>
<td align="right">0.595</td>
</tr>
<tr class="odd">
<td align="left">Gprimest</td>
<td align="right">0.632</td>
</tr>
</tbody>
</table>
<p>Another way to summarize data is to use violin plots.</p>
<pre class="r"><code>library(reshape2)
library(ggplot2)

dpf &lt;- melt(myDiff[,c(3:8,19)], varnames=c(&#39;Index&#39;, &#39;Sample&#39;), value.name = &#39;Depth&#39;, na.rm=TRUE)</code></pre>
<pre><code>## No id variables; using all as measure variables</code></pre>
<pre class="r"><code>p &lt;- ggplot(dpf, aes(x=variable, y=Depth)) + geom_violin(fill=&quot;#2ca25f&quot;, adjust = 1.2)
p &lt;- p + xlab(&quot;&quot;)
p &lt;- p + ylab(&quot;&quot;)
p &lt;- p + theme_bw()
p</code></pre>
<p><img src="genetic_differentiation_files/figure-html/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;" /></p>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-hedrick2005standardized" class="csl-entry">
Hedrick, Philip W. 2005. <span>“A Standardized Genetic Differentiation
Measure.”</span> <em>Evolution</em> 59 (8): 1633–38.
</div>
<div id="ref-jost2008gst" class="csl-entry">
Jost, Lou. 2008. <span>“GST and Its Relatives Do Not Measure
Differentiation.”</span> <em>Molecular Ecology</em> 17 (18): 4015–26.
</div>
<div id="ref-nei1973analysis" class="csl-entry">
Nei, Masatoshi. 1973. <span>“Analysis of Gene Diversity in Subdivided
Populations.”</span> <em>Proceedings of the National Academy of
Sciences</em> 70 (12): 3321–23.
</div>
<div id="ref-simpson1949measurement" class="csl-entry">
Simpson, Edward H. 1949. <span>“Measurement of Diversity.”</span>
<em>Nature</em> 163: 688.
</div>
<div id="ref-wright1949genetical" class="csl-entry">
Wright, Sewall. 1949. <span>“The Genetical Structure of
Populations.”</span> <em>Annals of Eugenics</em> 15 (1): 323–54.
</div>
</div>
</div>

<center>
<hr class="style1">
<p>Copyright &copy; 2017, 2018 Brian J. Knaus. All rights reserved.</p>
<p>USDA Agricultural Research Service, Horticultural Crops Research Lab.</p>
</center>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
