<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Extracting matrices from vcfR objects</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-2"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-2');
</script>


<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-1"></script>
<script> 
window.dataLayer = window.dataLayer || []; 
function gtag(){dataLayer.push(arguments)}; 
gtag('js', new Date()); 
gtag('config', 'UA-107144798-1'); 
</script>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">vcfR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="rlanguage.html">R language</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="quick_intro.html">A quick introduction</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="vcfR_object.html">vcfR objects</a>
    </li>
    <li>
      <a href="how_much_memory.html">How much memory</a>
    </li>
    <li>
      <a href="matrices.html">Extracting matrices</a>
    </li>
    <li>
      <a href="tidy_vcfR.html">Tidy vcfR</a>
    </li>
    <li>
      <a href="chromR_object.html">chromR objects</a>
    </li>
    <li>
      <a href="visualization_1.html">Visualization 1</a>
    </li>
    <li>
      <a href="visualization_2.html">Visualization 2</a>
    </li>
    <li>
      <a href="sequence_coverage.html">Sequence coverage</a>
    </li>
    <li>
      <a href="filtering_data.html">Filtering data</a>
    </li>
    <li>
      <a href="ranking_data.html">Ranking data</a>
    </li>
    <li>
      <a href="windowing.html">Windowing</a>
    </li>
    <li>
      <a href="genetic_differentiation.html">Genetic differentiation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    GBS class
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="gbs_class.html">GBS class</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="extract_data.html">Extract data</a>
    </li>
    <li>
      <a href="depth_plot.html">Depth plot</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="censoring_data.html">Censoring data</a>
    </li>
    <li>
      <a href="omitting_data.html">Omitting data</a>
    </li>
    <li>
      <a href="apply.html">Apply</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Ploidy
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ploidy.html">Ploidy</a>
    </li>
    <li>
      <a href="determining_ploidy_1.html">Determining ploidy 1</a>
    </li>
    <li>
      <a href="determining_ploidy_2.html">Determining ploidy 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Export
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="export.html">Overview</a>
    </li>
    <li>
      <a href="export_vcfgz.html">Export to *vcf.gz</a>
    </li>
    <li>
      <a href="export_genind_genclone.html">Genind and Genclone</a>
    </li>
    <li>
      <a href="export_genlight_snpclone.html">Genlight and SNPclone</a>
    </li>
    <li>
      <a href="dnabin.html">DNAbin</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="subset_data_to_1chrom.html">Subset to 1 chromosome</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="vcf_software.html">VCF software</a>
    </li>
    <li>
      <a href="dip_to_hap.html">Haploidizing diploid data</a>
    </li>
    <li>
      <a href="compiling_vcfR.html">Compiling vcfR</a>
    </li>
    <li>
      <a href="reporting_issue.html">Reporting an issue</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<p>
<center>
<h3>vcfR documentation</h3>
by
<br>
Brian J. Knaus and Niklaus J. Gr&uuml;nwald
</center>
</p>




<div id="header">



<h1 class="title toc-ignore">Extracting matrices from vcfR objects</h1>

</div>


<p>The data contained in VCF files presents a challenge to analysis in
that the data are not strictly tabular and are large. Because the data
are not strictly tabular we need to find ways to coerce the data into
tables or matrices. Once the data is stored as a matrix it can be
operated on by numerous R functions and packages. Because the data are
large, this process needs to happen in a computationally efficent
manner. Here we explore how vcfR facilitates these tasks.</p>
<div id="extracting-data-matrices" class="section level2">
<h2>Extracting data matrices</h2>
<p>Typical VCF datasets are so large that it is not practical to view
them in their entirety. Here we use the example dataset
<code>vcfR_test</code> which comes with vcfR. This is a nice dataset
because it is small enough to visualize.</p>
<pre class="r"><code>library(vcfR)
data(&quot;vcfR_test&quot;)
head(vcfR_test)</code></pre>
<pre><code>## [1] &quot;***** Object of class &#39;vcfR&#39; *****&quot;
## [1] &quot;***** Meta section *****&quot;
## [1] &quot;##fileformat=VCFv4.3&quot;
## [1] &quot;##fileDate=20090805&quot;
## [1] &quot;##source=myImputationProgramV3.1&quot;
## [1] &quot;##reference=file:///seq/references/1000GenomesPilot-NCBI36.fasta&quot;
## [1] &quot;##contig=&lt;ID=20,length=62435964,assembly=B36,md5=f126cdf8a6e0c7f379d [Truncated]&quot;
## [1] &quot;##phasing=partial&quot;
## [1] &quot;First 6 rows.&quot;
## [1] 
## [1] &quot;***** Fixed section *****&quot;
##      CHROM POS       ID          REF   ALT      QUAL FILTER
## [1,] &quot;20&quot;  &quot;14370&quot;   &quot;rs6054257&quot; &quot;G&quot;   &quot;A&quot;      &quot;29&quot; &quot;PASS&quot;
## [2,] &quot;20&quot;  &quot;17330&quot;   NA          &quot;T&quot;   &quot;A&quot;      &quot;3&quot;  &quot;q10&quot; 
## [3,] &quot;20&quot;  &quot;1110696&quot; &quot;rs6040355&quot; &quot;A&quot;   &quot;G,T&quot;    &quot;67&quot; &quot;PASS&quot;
## [4,] &quot;20&quot;  &quot;1230237&quot; NA          &quot;T&quot;   NA       &quot;47&quot; &quot;PASS&quot;
## [5,] &quot;20&quot;  &quot;1234567&quot; &quot;microsat1&quot; &quot;GTC&quot; &quot;G,GTCT&quot; &quot;50&quot; &quot;PASS&quot;
## [1] 
## [1] &quot;***** Genotype section *****&quot;
##      FORMAT        NA00001          NA00002          NA00003       
## [1,] &quot;GT:GQ:DP:HQ&quot; &quot;0|0:48:1:51,51&quot; &quot;1|0:48:8:51,51&quot; &quot;1/1:43:5:.,.&quot;
## [2,] &quot;GT:GQ:DP:HQ&quot; &quot;0|0:49:3:58,50&quot; &quot;0|1:3:5:65,3&quot;   &quot;0/0:41:3&quot;    
## [3,] &quot;GT:GQ:DP:HQ&quot; &quot;1|2:21:6:23,27&quot; &quot;2|1:2:0:18,2&quot;   &quot;2/2:35:4&quot;    
## [4,] &quot;GT:GQ:DP:HQ&quot; &quot;0|0:54:7:56,60&quot; &quot;0|0:48:4:51,51&quot; &quot;0/0:61:2&quot;    
## [5,] &quot;GT:GQ:DP&quot;    &quot;0/1:35:4&quot;       &quot;0/2:17:2&quot;       &quot;1/1:40:3&quot;    
## [1] 
## [1] &quot;Unique GT formats:&quot;
## [1] &quot;GT:GQ:DP:HQ&quot; &quot;GT:GQ:DP&quot;   
## [1]</code></pre>
<p>In the genotype section we see four columns: a FORMAT column that
indicates the contents of subsequent columns and three sample columns.
Variants one through four contain four pieces of information with each
genotype (GT:GQ:DP:HQ). The meaning of these acronyms should be defined
in the meta section. Each element of the genotype matrix therefore
contains a colon delimeted list that we need to parse before we can work
with it. For example, if we wanted to analyze teh genotypes we would
need to isolate them from the other information. Thsi can be
accomplished using the function <code>extract.gt()</code>.</p>
<pre class="r"><code>gt &lt;- extract.gt(vcfR_test)
gt</code></pre>
<pre><code>##            NA00001 NA00002 NA00003
## rs6054257  &quot;0|0&quot;   &quot;1|0&quot;   &quot;1/1&quot;  
## 20_17330   &quot;0|0&quot;   &quot;0|1&quot;   &quot;0/0&quot;  
## rs6040355  &quot;1|2&quot;   &quot;2|1&quot;   &quot;2/2&quot;  
## 20_1230237 &quot;0|0&quot;   &quot;0|0&quot;   &quot;0/0&quot;  
## microsat1  &quot;0/1&quot;   &quot;0/2&quot;   &quot;1/1&quot;</code></pre>
<p>When a genotype section is included in VCF data the one mandatory
field is the genotype (GT) and it must be the first. The genotype is the
default field for <code>extract.gt()</code> to work on. Note that the
samplers names have been used to name the columns. The variant names
(row names) are a little more complicated. The ‘ID’ column of the fixed
section contains optional unique identifiers. These typically come from
some sort of database of variants. If this column were entirely
populated with unique identifiers we could use it to name our variants.
However, all variants do not always have an ID. For example, variants 2
and 4 have missing IDs (NA). The function <code>extract.gt()</code>
tries to name these variants by combining the chromosome (CHROM) and the
position (POS) with the delimiter ’_’ to create a name. We see in our
matrix of genotypes that they all have a name.</p>
<p>We can also extract numeric data from our VCF data. We can do this by
selecting a numeric element and setting the parameter
<code>as.numeric = TRUE</code>.</p>
<pre class="r"><code>gt &lt;- extract.gt(vcfR_test, element = &#39;DP&#39;, as.numeric = TRUE)
gt</code></pre>
<pre><code>##            NA00001 NA00002 NA00003
## rs6054257        1       8       5
## 20_17330         3       5       3
## rs6040355        6       0       4
## 20_1230237       7       4       2
## microsat1        4       2       3</code></pre>
<p>Here we’ve extracted the sequence depth (DP) of each variant and
converted it to a numeric. Because they are numeric they may be used for
quantitative comparisons. Much of R is designed to work with matrices of
numbers. Be careful setting the parameter
<code>as.numeric = TRUE</code>. It will do its best to convert the data
to a numeric, even if this may not make sense. For example, we could
convert the genotype to numeric.</p>
<pre class="r"><code>gt &lt;- extract.gt(vcfR_test, element = &#39;GT&#39;, as.numeric = TRUE)
gt</code></pre>
<pre><code>##            NA00001 NA00002 NA00003
## rs6054257        0       1       1
## 20_17330         0       0       0
## rs6040355        1       2       2
## 20_1230237       0       0       0
## microsat1        0       0       1</code></pre>
<p>This operation did not throw an error, but the resulting matrix
doesn’t appear to make much sense. Only ask to convert numeric data to
numeric data or you will probably get unexpected results.</p>
</div>
<div id="matrix-parsing" class="section level2">
<h2>Matrix parsing</h2>
<p>Some elements fromthe genotype section may require further parsing.
In our example the haplotype quality (HQ) includes a quality for each
haplotype and these values are comma delimited.</p>
<pre class="r"><code>gt &lt;- extract.gt(vcfR_test, element = &#39;HQ&#39;)
gt</code></pre>
<pre><code>##            NA00001 NA00002 NA00003
## rs6054257  &quot;51,51&quot; &quot;51,51&quot; &quot;.,.&quot;  
## 20_17330   &quot;58,50&quot; &quot;65,3&quot;  NA     
## rs6040355  &quot;23,27&quot; &quot;18,2&quot;  NA     
## 20_1230237 &quot;56,60&quot; &quot;51,51&quot; NA     
## microsat1  NA      NA      NA</code></pre>
<p>Note that some elements contain data while in others it is
missing.</p>
<pre class="r"><code>myHQ1 &lt;- masplit(gt[,1:2], sort = 0)
myHQ1</code></pre>
<pre><code>##            NA00001 NA00002
## rs6054257       51      51
## 20_17330        58      65
## rs6040355       23      18
## 20_1230237      56      51
## microsat1       NA      NA</code></pre>
<p>The function <code>masplit()</code> is fairly flexible in that you
can choose which element to return and you can also sort the data prior
to selection. These functions should help make VCF data accessible to
the wealth of existing R functions and packages.</p>
</div>

<center>
<hr class="style1">
<p>Copyright &copy; 2017, 2018 Brian J. Knaus. All rights reserved.</p>
<p>USDA Agricultural Research Service, Horticultural Crops Research Lab.</p>
</center>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
