<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Missing data</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-2"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-2');
</script>


<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-1"></script>
<script> 
window.dataLayer = window.dataLayer || []; 
function gtag(){dataLayer.push(arguments)}; 
gtag('js', new Date()); 
gtag('config', 'UA-107144798-1'); 
</script>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">vcfR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="rlanguage.html">R language</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="quick_intro.html">A quick introduction</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="vcfR_object.html">vcfR objects</a>
    </li>
    <li>
      <a href="how_much_memory.html">How much memory</a>
    </li>
    <li>
      <a href="matrices.html">Extracting matrices</a>
    </li>
    <li>
      <a href="tidy_vcfR.html">Tidy vcfR</a>
    </li>
    <li>
      <a href="chromR_object.html">chromR objects</a>
    </li>
    <li>
      <a href="visualization_1.html">Visualization 1</a>
    </li>
    <li>
      <a href="visualization_2.html">Visualization 2</a>
    </li>
    <li>
      <a href="sequence_coverage.html">Sequence coverage</a>
    </li>
    <li>
      <a href="filtering_data.html">Filtering data</a>
    </li>
    <li>
      <a href="ranking_data.html">Ranking data</a>
    </li>
    <li>
      <a href="windowing.html">Windowing</a>
    </li>
    <li>
      <a href="genetic_differentiation.html">Genetic differentiation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    GBS class
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="gbs_class.html">GBS class</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="extract_data.html">Extract data</a>
    </li>
    <li>
      <a href="depth_plot.html">Depth plot</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="censoring_data.html">Censoring data</a>
    </li>
    <li>
      <a href="omitting_data.html">Omitting data</a>
    </li>
    <li>
      <a href="apply.html">Apply</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Ploidy
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ploidy.html">Ploidy</a>
    </li>
    <li>
      <a href="determining_ploidy_1.html">Determining ploidy 1</a>
    </li>
    <li>
      <a href="determining_ploidy_2.html">Determining ploidy 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Export
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="export.html">Overview</a>
    </li>
    <li>
      <a href="export_vcfgz.html">Export to *vcf.gz</a>
    </li>
    <li>
      <a href="export_genind_genclone.html">Genind and Genclone</a>
    </li>
    <li>
      <a href="export_genlight_snpclone.html">Genlight and SNPclone</a>
    </li>
    <li>
      <a href="dnabin.html">DNAbin</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="subset_data_to_1chrom.html">Subset to 1 chromosome</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="vcf_software.html">VCF software</a>
    </li>
    <li>
      <a href="dip_to_hap.html">Haploidizing diploid data</a>
    </li>
    <li>
      <a href="compiling_vcfR.html">Compiling vcfR</a>
    </li>
    <li>
      <a href="reporting_issue.html">Reporting an issue</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<p>
<center>
<h3>vcfR documentation</h3>
by
<br>
Brian J. Knaus and Niklaus J. Gr&uuml;nwald
</center>
</p>




<div id="header">



<h1 class="title toc-ignore">Missing data</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#overall-missingness" id="toc-overall-missingness">Overall
missingness</a></li>
<li><a href="#quantifying-missingness-one-sample"
id="toc-quantifying-missingness-one-sample">Quantifying missingness, one
sample</a></li>
<li><a href="#quantifying-missingness-all-samples"
id="toc-quantifying-missingness-all-samples">Quantifying missingness,
all samples</a></li>
</ul>
</div>

<p>As the size of our dataset grow in terms of samples and variants the
size of our data matrix grows. As the size of our data matrix grows it
also increases the opportunity to have missing data. Also, some of our
quality filtering steps increased the degree of missingness in our data
matrix by setting values that we determined to be of unusual quality to
NA. One way of managing missing data is to use imputation, a set of
methods that attempts to infer what the most likely genotype should be
and replaces the missing genotype with the interpolated genotype.
However, if your data has a large degree of missingness you may want to
attempt to mitigate missingness instead of interpolation Or you may want
to implement a mitigation step prior to interpolation in the hope that
this will improve the performance of the interpolation. Missing data can
frequently be due to samples (columns) or variants (rows) of low
quality. Here we demonstrate how to identify samples and variants in the
data set that have a high degree of missingness. In another section
we’ll show how to omit them.</p>
<div id="overall-missingness" class="section level2">
<h2>Overall missingness</h2>
<p>A first perspective on how complete our dataset is can be provided by
the <code>show</code> method for the vcfR object. When you invoke the
name of an object with no arguments it invokes the show method.</p>
<pre class="r"><code>vcf</code></pre>
<pre><code>## ***** Object of Class vcfR *****
## 61 samples
## 7171 CHROMs
## 69,296 variants
## Object size: 47.7 Mb
## 37.62 percent missing data
## *****        *****         *****</code></pre>
<p>The show method for an object typically reports a summary of what is
contained in the object. Here we see the number of samples and variants
in our data. We also see a report of what the percentage of missing data
is in our object. In the context of vcfR this is the proportion of
variants scored as NA. Note that if a variant includes some data
associated with a missing genotype it will not be recognized as missing.
For example, a missing genotype could be associated with a depth
information as follows.</p>
<pre class="r"><code>GT:DP ./.:1</code></pre>
<p>Because this variant does include some data it will not be recognized
as missing until the genotypes are extracted and queried for
missingness. This means that the degree of missingness reported by the
<code>show</code> method may be an underrepresentation. It does provide
an easily accessed first perspective on the proportion of missing data.
To determine the cause of this missing data (e.g., are there particular
samples or variants of poor quality) we will look further.</p>
</div>
<div id="quantifying-missingness-one-sample" class="section level2">
<h2>Quantifying missingness, one sample</h2>
<p>To quantify missingness for a single sample we can use the function
<code>is.na()</code>. This function uses a vector as an argument and
returns a logical vector (TRUE and FALSE) indicating which values are
NA. If we remind ourselves that TRUEs and FALSEs are numerically encoded
as ones and zeros it reminds us we can take a sum of this logical vector
to determine the degree of missingness.</p>
<pre class="r"><code>as.numeric(TRUE)</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code>sum(is.na(dp[,1]))</code></pre>
<pre><code>## [1] 11725</code></pre>
<p>This reports the number of missing variants in the first sample. We
could similarly count the number of missing samples from a variant by
accessing a row instead of a column. We could also convert this to a
percentage by using <code>length()</code> to determine the total number
of values in either the column or row and use this as a denominator.</p>
</div>
<div id="quantifying-missingness-all-samples" class="section level2">
<h2>Quantifying missingness, all samples</h2>
<p>This is illustrative of what we can accomplish for a single sample or
variant. We typically have many samples an tens of thousand (or more)
variants. We can extend the functionality of the above example to many
columns or rows by using the <code>apply()</code> function. See the
section on apply if you are unfamiliar with this function. Because we
will be summarizing many samples we will use a barplot to visualize the
results as opposed to trying to scrutinize the numerical
information.</p>
<pre class="r"><code>myMiss &lt;- apply(dp, MARGIN = 2, function(x){ sum(is.na(x)) })
myMiss &lt;- myMiss/nrow(vcf)

library(RColorBrewer)
palette(brewer.pal(n=12, name = &#39;Set3&#39;))

par(mar = c(12,4,4,2))
barplot(myMiss, las = 2, col = 1:12)
title(ylab = &quot;Missingness (%)&quot;)</code></pre>
<p><img src="missing_data_files/figure-html/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mar = c(5,4,4,2))</code></pre>
<p>We see that there are two classes of samples. The samples with long
names have a high degree of missingness (around 80%) while the samples
with short names have a relatively low degree of missingness.</p>
<p>We can do something similar to query the variants (rows) for
missingness. However, our dataset has over 60 thousand variants, so we
wouldn’t want to visualize this with a barchart. It would require a
barchart with 60 thousand bars. Instead of usung a barchart we’ll use a
histogram.</p>
<pre class="r"><code>myMiss &lt;- apply(dp, MARGIN = 1, function(x){ sum(is.na(x)) })
myMiss &lt;- myMiss/ncol(vcf@gt[,-1])

hist(myMiss, col = &quot;#8DD3C7&quot;, xlab = &quot;Missingness (%)&quot;)</code></pre>
<p><img src="missing_data_files/figure-html/unnamed-chunk-6-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>We’ve now seen how we can create summaries of our data matrix over
both rows and columns. Once we have this knowlege in hand we may develop
a plan for managing this data. Future sections will provide examples of
mitigating data that has been determined to be undesireable.</p>
</div>

<center>
<hr class="style1">
<p>Copyright &copy; 2017, 2018 Brian J. Knaus. All rights reserved.</p>
<p>USDA Agricultural Research Service, Horticultural Crops Research Lab.</p>
</center>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
