<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Quality filtering</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-2"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-2');
</script>


<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-1"></script>
<script> 
window.dataLayer = window.dataLayer || []; 
function gtag(){dataLayer.push(arguments)}; 
gtag('js', new Date()); 
gtag('config', 'UA-107144798-1'); 
</script>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">vcfR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="rlanguage.html">R language</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="quick_intro.html">A quick introduction</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="vcfR_object.html">vcfR objects</a>
    </li>
    <li>
      <a href="how_much_memory.html">How much memory</a>
    </li>
    <li>
      <a href="matrices.html">Extracting matrices</a>
    </li>
    <li>
      <a href="tidy_vcfR.html">Tidy vcfR</a>
    </li>
    <li>
      <a href="chromR_object.html">chromR objects</a>
    </li>
    <li>
      <a href="visualization_1.html">Visualization 1</a>
    </li>
    <li>
      <a href="visualization_2.html">Visualization 2</a>
    </li>
    <li>
      <a href="sequence_coverage.html">Sequence coverage</a>
    </li>
    <li>
      <a href="filtering_data.html">Filtering data</a>
    </li>
    <li>
      <a href="ranking_data.html">Ranking data</a>
    </li>
    <li>
      <a href="windowing.html">Windowing</a>
    </li>
    <li>
      <a href="genetic_differentiation.html">Genetic differentiation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    GBS class
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="gbs_class.html">GBS class</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="extract_data.html">Extract data</a>
    </li>
    <li>
      <a href="depth_plot.html">Depth plot</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="censoring_data.html">Censoring data</a>
    </li>
    <li>
      <a href="omitting_data.html">Omitting data</a>
    </li>
    <li>
      <a href="apply.html">Apply</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Ploidy
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ploidy.html">Ploidy</a>
    </li>
    <li>
      <a href="determining_ploidy_1.html">Determining ploidy 1</a>
    </li>
    <li>
      <a href="determining_ploidy_2.html">Determining ploidy 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Export
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="export.html">Overview</a>
    </li>
    <li>
      <a href="export_vcfgz.html">Export to *vcf.gz</a>
    </li>
    <li>
      <a href="export_genind_genclone.html">Genind and Genclone</a>
    </li>
    <li>
      <a href="export_genlight_snpclone.html">Genlight and SNPclone</a>
    </li>
    <li>
      <a href="dnabin.html">DNAbin</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="subset_data_to_1chrom.html">Subset to 1 chromosome</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="vcf_software.html">VCF software</a>
    </li>
    <li>
      <a href="dip_to_hap.html">Haploidizing diploid data</a>
    </li>
    <li>
      <a href="compiling_vcfR.html">Compiling vcfR</a>
    </li>
    <li>
      <a href="reporting_issue.html">Reporting an issue</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<p>
<center>
<h3>vcfR documentation</h3>
by
<br>
Brian J. Knaus and Niklaus J. Gr&uuml;nwald
</center>
</p>




<div id="header">



<h1 class="title toc-ignore">Quality filtering</h1>

</div>


<p>Once a sequencing run has been completed the sequence reads are
typically aligned to a reference genome and variants are called. These
variants are typically stored in a VCF format file. THe R package vcfR
was designed to import and export this data into the R environment. Once
in the R environment, vcfR provides tools to manipulate the data to help
you understand its content. Common issues in variant data may be that
certain samples may be of low quality or certain vriants may be of
questionable quality. Here we’ll provide examples of how vcfR can help
accomplish these tasks.</p>
<div id="data-import" class="section level2">
<h2>Data import</h2>
<p>Our first step will be to import the VCF data into R. The result will
be that our VCF data will be stored in a ‘vcfR’ object. Once we have
read it in we can validate our data is what we expect by using the
<code>show</code> method. The <code>show</code> method is available for
many types of objects and is implemented when the object’s name is
entered at the console with no other information. THe <code>show</code>
method typically provides summary information about the object and what
it contains.</p>
<pre class="r"><code>library(vcfR)
vcf &lt;- read.vcfR(&#39;TASSEL_GBS0077.vcf.gz&#39;)
class(vcf)</code></pre>
<pre class="r"><code>vcf</code></pre>
<pre><code>## ***** Object of Class vcfR *****
## 61 samples
## 7171 CHROMs
## 69,296 variants
## Object size: 47.7 Mb
## 37.62 percent missing data
## *****        *****         *****</code></pre>
<p>The results from the <code>show</code> method informs us that our
vcfR object contains 61 samples and 69,296 variants. This is what we
expected so we can be confident that our data was imported correctly.
The output also informs us that there is just over 37% missing data in
our file. In our experience this is typical for GBS data sets.</p>
</div>
<div id="extract-depth-dp" class="section level2">
<h2>Extract depth (DP)</h2>
<p>The vcfR function <code>extract.gt()</code> is used to extract
matrices of data from teh GT portion of VCF data. the funtion
<code>extract.gt()</code> provides a link between VCF data and R. Much
of R is designed to operate on matrices of data and once
<code>extract.gt()</code> provides this matrix the universe of R becomes
available. Note that we use the ‘as.numeric=TRUE’ option here. We should
only use this option when we are certain that we have numeric data. If
you use it on non-numeric data R will do its best to do something, which
is not likely to be what you expect. We can use the
<code>queryMETA()</code> function remind us what this element is.</p>
<pre class="r"><code>queryMETA(vcf, element = &#39;FORMAT.+DP&#39;)</code></pre>
<pre><code>## [[1]]
## [1] &quot;FORMAT=ID=DP&quot;                                                 
## [2] &quot;Number=1&quot;                                                     
## [3] &quot;Type=Integer&quot;                                                 
## [4] &quot;Description=Read Depth (only filtered reads used for calling)&quot;</code></pre>
<pre class="r"><code>vcf@gt[1:4,1:4]</code></pre>
<pre><code>##      FORMAT           1                       10                       
## [1,] &quot;GT:AD:DP:GQ:PL&quot; &quot;0/0:5,0:5:96:0,15,180&quot; &quot;1/1:0,10:10:99:255,30,0&quot;
## [2,] &quot;GT:AD:DP:GQ:PL&quot; &quot;1/1:0,1:1:66:36,3,0&quot;   &quot;0/0:4,0:4:94:0,12,144&quot;  
## [3,] &quot;GT:AD:DP:GQ:PL&quot; NA                      NA                       
## [4,] &quot;GT:AD:DP:GQ:PL&quot; &quot;1/1:0,5:5:96:180,15,0&quot; &quot;0/0:8,0:8:99:0,24,255&quot;  
##      13                     
## [1,] &quot;0/0:9,0:9:99:0,27,255&quot;
## [2,] &quot;0/0:1,0:1:66:0,3,36&quot;  
## [3,] &quot;0/0:3,0:3:88:0,9,108&quot; 
## [4,] NA</code></pre>
<pre class="r"><code>dp &lt;- extract.gt(vcf, element = &quot;DP&quot;, as.numeric=TRUE)
dp[1:4,1:3]</code></pre>
<pre><code>##          1 10 13
## S1_4509  5 10  9
## S1_4657  1  4  1
## S1_5193 NA NA  3
## S1_5647  5  8 NA</code></pre>
</div>
<div id="missing-data" class="section level2">
<h2>Missing data</h2>
<pre class="r"><code>sum(is.na(dp[,1]))</code></pre>
<pre><code>## [1] 11725</code></pre>
<pre class="r"><code># apply(dp, MARGIN = 2, function(x){sum(is.na(x))})</code></pre>
<pre class="r"><code>myMiss &lt;- apply(dp, MARGIN = 2, function(x){sum(is.na(x))})
myMiss &lt;- myMiss/nrow(vcf)

library(RColorBrewer)
palette(brewer.pal(n=12, name = &#39;Set3&#39;))

par(mar = c(12,4,4,2))
barplot(myMiss, las = 2, col = 1:12)
title(ylab = &quot;Missingness&quot;)</code></pre>
<p><img src="quality_filtering_files/figure-html/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mar = c(5,4,4,2))</code></pre>
</div>
<div id="sequence-depth" class="section level2">
<h2>Sequence depth</h2>
<pre class="r"><code>boxplot(dp, col=2:8, las=3)</code></pre>
<p><img src="quality_filtering_files/figure-html/unnamed-chunk-6-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>library(reshape2)
library(ggplot2) 
library(cowplot)

# Melt our matrix into a long form data.frame.
dpf &lt;- melt(dp, varnames=c(&#39;Index&#39;, &#39;Sample&#39;), value.name = &#39;Depth&#39;, na.rm=TRUE)
dpf &lt;- dpf[ dpf$Depth &gt; 0,]

# Create a row designator.
# You may want to adjust this
#samps_per_row &lt;- 20
samps_per_row &lt;- 16
myRows &lt;- ceiling(length(levels(dpf$Sample))/samps_per_row)
myList &lt;- vector(mode = &quot;list&quot;, length = myRows)

for(i in 1:myRows){
  myIndex &lt;- c(i*samps_per_row - samps_per_row + 1):c(i*samps_per_row)
  myIndex &lt;- myIndex[myIndex &lt;= length(levels(dpf$Sample))]
  myLevels &lt;- levels(dpf$Sample)[myIndex]
  myRegex &lt;- paste(myLevels, collapse = &quot;$|^&quot;)
  myRegex &lt;- paste(&quot;^&quot;, myRegex, &quot;$&quot;, sep = &quot;&quot;)
  myList[[i]] &lt;- dpf[grep(myRegex, dpf$Sample),]
  myList[[i]]$Sample &lt;- factor(myList[[i]]$Sample)
}

# Create the plot.
myPlots &lt;- vector(mode = &quot;list&quot;, length = myRows)
for(i in 1:myRows){
  myPlots[[i]] &lt;- ggplot(myList[[i]], aes(x=Sample, y=Depth)) + 
                  geom_violin(fill=&quot;#90EE90&quot;, adjust=1.0, scale = &quot;count&quot;, trim=TRUE)

  myPlots[[i]] &lt;- myPlots[[i]] + theme_bw()
  myPlots[[i]] &lt;- myPlots[[i]] + theme(axis.title.x = element_blank(), 
                  axis.text.x = element_text(angle = 60, hjust = 1))
  myPlots[[i]] &lt;- myPlots[[i]] + scale_y_continuous(trans=scales::log2_trans(), 
                  breaks=c(1, 10, 100, 800),
                  minor_breaks=c(1:10, 2:10*10, 2:8*100))
  myPlots[[i]] &lt;- myPlots[[i]] + theme( panel.grid.major.y=element_line(color = &quot;#A9A9A9&quot;, size=0.6) )
  myPlots[[i]] &lt;- myPlots[[i]] + theme( panel.grid.minor.y=element_line(color = &quot;#C0C0C0&quot;, size=0.2) )
}</code></pre>
<pre class="r"><code># Plot the plot.
plot_grid(plotlist = myPlots, nrow = myRows)</code></pre>
<p><img src="quality_filtering_files/figure-html/unnamed-chunk-8-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>Once we have extracted the quantile information we can use apply and
quantile to build intervals for what we may consider acceptable
coverage.</p>
<pre class="r"><code>quants &lt;- apply(dp, MARGIN=2, quantile, probs=c(0.1, 0.8), na.rm=TRUE)
#quants &lt;- apply(dp, MARGIN=2, quantile, probs=c(0.34, 0.68), na.rm=TRUE)
quants[,1:17]</code></pre>
<pre><code>##      1 10 13 1725-d-8-21-07-r-1-pm-r 18 1858-d-10-15-07-p-3-pm-r  2 20 22 31 34
## 10%  1  2  1                       1  1                        1  2  2  2  2  1
## 80% 18 42 17                      15 25                       16 34 31 37 29 26
##     3450-d-4-06-09-r-3-pm-r 37 39  4 40 43
## 10%                       1  2  2  2  2  1
## 80%                      12 27 36 27 33 23</code></pre>
<p>We can now use these thresholds to censor data outside this
threshold.</p>
<pre class="r"><code>dp2 &lt;- sweep(dp, MARGIN=2, FUN = &quot;-&quot;, quants[1,])
dp[dp2 &lt; 0] &lt;- NA

dp2 &lt;- sweep(dp, MARGIN=2, FUN = &quot;-&quot;, quants[2,])
dp[dp2 &gt; 0] &lt;- NA

dp[dp &lt; 4] &lt;- NA</code></pre>
<p>Update the vcfR object with the censored data.</p>
<pre class="r"><code>vcf@gt[,-1][ is.na(dp) == TRUE ] &lt;- NA</code></pre>
<p>We’ll want to see how this has affected the missingness of our vcfR
object.</p>
<pre class="r"><code>vcf</code></pre>
<pre><code>## ***** Object of Class vcfR *****
## 61 samples
## 7171 CHROMs
## 69,296 variants
## Object size: 46.3 Mb
## 66.63 percent missing data
## *****        *****         *****</code></pre>
<p>We’ll want to mitigate variants with a high degree of
missingness.</p>
<pre class="r"><code>dp &lt;- extract.gt(vcf, element = &quot;DP&quot;, as.numeric=TRUE)

miss &lt;- apply(dp, MARGIN=1, function(x){sum(is.na(x))})
miss &lt;- miss/ncol(dp)</code></pre>
<p>Plot a histogram.</p>
<pre class="r"><code>hist(miss, col=8, breaks=seq(0,1,by=0.02))</code></pre>
<p><img src="quality_filtering_files/figure-html/unnamed-chunk-14-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>Omit variants with a high degree of missingness.</p>
<pre class="r"><code>#vcf &lt;- vcf[miss &lt; 0.05,]
vcf &lt;- vcf[miss &lt; 0.1,]
vcf</code></pre>
<pre><code>## ***** Object of Class vcfR *****
## 61 samples
## 102 CHROMs
## 295 variants
## Object size: 0.2 Mb
## 7.719 percent missing data
## *****        *****         *****</code></pre>
<pre class="r"><code>dp &lt;- extract.gt(vcf, element = &quot;DP&quot;, as.numeric=TRUE)
heatmap.bp(dp, rlabels = FALSE)</code></pre>
<p><img src="quality_filtering_files/figure-html/unnamed-chunk-16-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#heatmap.bp(dp[1:1000,], rlabels = FALSE)</code></pre>
<pre class="r"><code>boxplot(dp, col=2:8, las=3)</code></pre>
<p><img src="quality_filtering_files/figure-html/unnamed-chunk-17-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="output-to-file" class="section level2">
<h2>Output to file</h2>
<pre class="r"><code>write.vcf(tas, &#39;TASSEL_GBS0077_dp_filtered.vcf.gz&#39;)</code></pre>
</div>

<center>
<hr class="style1">
<p>Copyright &copy; 2017, 2018 Brian J. Knaus. All rights reserved.</p>
<p>USDA Agricultural Research Service, Horticultural Crops Research Lab.</p>
</center>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
