<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Visualizing VCF data 2</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-2"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-2');
</script>


<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-1"></script>
<script> 
window.dataLayer = window.dataLayer || []; 
function gtag(){dataLayer.push(arguments)}; 
gtag('js', new Date()); 
gtag('config', 'UA-107144798-1'); 
</script>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">vcfR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="rlanguage.html">R language</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="quick_intro.html">A quick introduction</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="vcfR_object.html">vcfR objects</a>
    </li>
    <li>
      <a href="how_much_memory.html">How much memory</a>
    </li>
    <li>
      <a href="matrices.html">Extracting matrices</a>
    </li>
    <li>
      <a href="tidy_vcfR.html">Tidy vcfR</a>
    </li>
    <li>
      <a href="chromR_object.html">chromR objects</a>
    </li>
    <li>
      <a href="visualization_1.html">Visualization 1</a>
    </li>
    <li>
      <a href="visualization_2.html">Visualization 2</a>
    </li>
    <li>
      <a href="sequence_coverage.html">Sequence coverage</a>
    </li>
    <li>
      <a href="filtering_data.html">Filtering data</a>
    </li>
    <li>
      <a href="ranking_data.html">Ranking data</a>
    </li>
    <li>
      <a href="windowing.html">Windowing</a>
    </li>
    <li>
      <a href="genetic_differentiation.html">Genetic differentiation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    GBS class
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="gbs_class.html">GBS class</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="extract_data.html">Extract data</a>
    </li>
    <li>
      <a href="depth_plot.html">Depth plot</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="censoring_data.html">Censoring data</a>
    </li>
    <li>
      <a href="omitting_data.html">Omitting data</a>
    </li>
    <li>
      <a href="apply.html">Apply</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Ploidy
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ploidy.html">Ploidy</a>
    </li>
    <li>
      <a href="determining_ploidy_1.html">Determining ploidy 1</a>
    </li>
    <li>
      <a href="determining_ploidy_2.html">Determining ploidy 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Export
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="export.html">Overview</a>
    </li>
    <li>
      <a href="export_vcfgz.html">Export to *vcf.gz</a>
    </li>
    <li>
      <a href="export_genind_genclone.html">Genind and Genclone</a>
    </li>
    <li>
      <a href="export_genlight_snpclone.html">Genlight and SNPclone</a>
    </li>
    <li>
      <a href="dnabin.html">DNAbin</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="subset_data_to_1chrom.html">Subset to 1 chromosome</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="vcf_software.html">VCF software</a>
    </li>
    <li>
      <a href="dip_to_hap.html">Haploidizing diploid data</a>
    </li>
    <li>
      <a href="compiling_vcfR.html">Compiling vcfR</a>
    </li>
    <li>
      <a href="reporting_issue.html">Reporting an issue</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<p>
<center>
<h3>vcfR documentation</h3>
by
<br>
Brian J. Knaus and Niklaus J. Gr&uuml;nwald
</center>
</p>




<div id="header">



<h1 class="title toc-ignore">Visualizing VCF data 2</h1>

</div>


<p>In the vignette ‘Visualizing VCF data I’ we began to explore how to
plot information contained in <a
href="http://samtools.github.io/hts-specs/"
title="VCF format at hts-specs">variant call format (vcf)</a> files.
This perspective was mostly one of summaries over all samples for each
variant. Here we build on this by exploring data based on each sample’s
genotype information.</p>
<div id="data" class="section level2">
<h2>Data</h2>
<p>As in other vignettes, we begin by loading the example data.</p>
<pre class="r"><code>library(vcfR)

vcf_file &lt;- system.file(&quot;extdata&quot;, &quot;pinf_sc50.vcf.gz&quot;, package = &quot;pinfsc50&quot;)
dna_file &lt;- system.file(&quot;extdata&quot;, &quot;pinf_sc50.fasta&quot;, package = &quot;pinfsc50&quot;)
gff_file &lt;- system.file(&quot;extdata&quot;, &quot;pinf_sc50.gff&quot;, package = &quot;pinfsc50&quot;)

vcf &lt;- read.vcfR(vcf_file, verbose = FALSE)
dna &lt;- ape::read.dna(dna_file, format = &quot;fasta&quot;)
gff &lt;- read.table(gff_file, sep=&quot;\t&quot;, quote=&quot;&quot;)

chrom &lt;- create.chromR(name=&quot;Supercontig&quot;, vcf=vcf, seq=dna, ann=gff, verbose=FALSE)
chrom &lt;- masker(chrom, min_DP = 300, max_DP = 700)
chrom &lt;- proc.chromR(chrom, verbose = FALSE)</code></pre>
</div>
<div id="genotype-data" class="section level2">
<h2>Genotype data</h2>
<p>In the vignette ‘Visualizing VCF data I’ we explored summaries of
variants made across all samples. If all samples are fairly homogenous
in the qualities of these metrics, than a summary across all samples may
be appropriate. However, if a few samples have very low quality, or some
samples are exceptional, this may be masked by looking at summaries
across all samples. By looking at the individual samples, these
differences may be discovered.</p>
<p>In order to look at the per sample information, we first need to
determine what sort of data we have there. We can use the function
head() to summarize the information in our Chrom object.</p>
<pre class="r"><code>head(chrom)</code></pre>
<pre><code>## *****   Class chromR, method head   *****
## Name:  Supercontig
## Length:  1,042,442
## 
## *****     Sample names (chromR)     *****
## [1] &quot;BL2009P4_us23&quot; &quot;DDR7602&quot;       &quot;IN2009T1_us22&quot; &quot;LBUS5&quot;        
## [5] &quot;NL07434&quot;       &quot;P10127&quot;       
## [1] &quot;...&quot;
## [1] &quot;P17777us22&quot;   &quot;P6096&quot;        &quot;P7722&quot;        &quot;RS2009P1_us8&quot; &quot;blue13&quot;      
## [6] &quot;t30-4&quot;       
## 
## *****    VCF fixed data (chromR)    *****
##      CHROM              POS   ID REF  ALT QUAL      FILTER
## [1,] &quot;Supercontig_1.50&quot; &quot;41&quot;  NA &quot;AT&quot; &quot;A&quot; &quot;4784.43&quot; NA    
## [2,] &quot;Supercontig_1.50&quot; &quot;136&quot; NA &quot;A&quot;  &quot;C&quot; &quot;550.27&quot;  NA    
## [3,] &quot;Supercontig_1.50&quot; &quot;254&quot; NA &quot;T&quot;  &quot;G&quot; &quot;774.44&quot;  NA    
## [4,] &quot;Supercontig_1.50&quot; &quot;275&quot; NA &quot;A&quot;  &quot;G&quot; &quot;714.53&quot;  NA    
## [5,] &quot;Supercontig_1.50&quot; &quot;386&quot; NA &quot;T&quot;  &quot;G&quot; &quot;876.55&quot;  NA    
## [6,] &quot;Supercontig_1.50&quot; &quot;462&quot; NA &quot;T&quot;  &quot;G&quot; &quot;1301.07&quot; NA    
## [1] &quot;...&quot;
##          CHROM              POS       ID REF  ALT QUAL      FILTER
## [22026,] &quot;Supercontig_1.50&quot; &quot;1042176&quot; NA &quot;T&quot;  &quot;A&quot; &quot;162.59&quot;  NA    
## [22027,] &quot;Supercontig_1.50&quot; &quot;1042196&quot; NA &quot;G&quot;  &quot;A&quot; &quot;180.86&quot;  NA    
## [22028,] &quot;Supercontig_1.50&quot; &quot;1042198&quot; NA &quot;T&quot;  &quot;G&quot; &quot;60.27&quot;   NA    
## [22029,] &quot;Supercontig_1.50&quot; &quot;1042303&quot; NA &quot;C&quot;  &quot;G&quot; &quot;804.15&quot;  NA    
## [22030,] &quot;Supercontig_1.50&quot; &quot;1042396&quot; NA &quot;GA&quot; &quot;G&quot; &quot;1578.82&quot; NA    
## [22031,] &quot;Supercontig_1.50&quot; &quot;1042398&quot; NA &quot;A&quot;  &quot;C&quot; &quot;1587.87&quot; NA    
## 
## INFO column has been suppressed, first INFO record:
##  [1] &quot;AC=32&quot;                   &quot;AF=1.00&quot;                
##  [3] &quot;AN=32&quot;                   &quot;DP=174&quot;                 
##  [5] &quot;FS=0.000&quot;                &quot;InbreedingCoeff=-0.0224&quot;
##  [7] &quot;MLEAC=32&quot;                &quot;MLEAF=1.00&quot;             
##  [9] &quot;MQ=51.30&quot;                &quot;MQ0=0&quot;                  
## [11] &quot;QD=27.50&quot;                &quot;SOR=4.103&quot;              
## 
## *****  VCF genotype data (chromR)  *****
## *****     First 6 columns      *********
##      FORMAT           BL2009P4_us23              DDR7602                    
## [1,] &quot;GT:AD:DP:GQ:PL&quot; &quot;1|1:0,7:7:21:283,21,0&quot;    &quot;1|1:0,6:6:18:243,18,0&quot;    
## [2,] &quot;GT:AD:DP:GQ:PL&quot; &quot;0|0:12,0:12:36:0,36,427&quot;  &quot;0|0:20,0:20:60:0,60,819&quot;  
## [3,] &quot;GT:AD:DP:GQ:PL&quot; &quot;0|0:27,0:27:81:0,81,1117&quot; &quot;0|0:26,0:26:78:0,78,1077&quot; 
## [4,] &quot;GT:AD:DP:GQ:PL&quot; &quot;0|0:29,0:29:87:0,87,1243&quot; &quot;0|0:27,0:27:81:0,81,1158&quot; 
## [5,] &quot;GT:AD:DP:GQ:PL&quot; &quot;0|0:26,0:26:78:0,78,1034&quot; &quot;0|0:30,0:30:90:0,90,1242&quot; 
## [6,] &quot;GT:AD:DP:GQ:PL&quot; &quot;0|0:23,0:23:69:0,69,958&quot;  &quot;0|0:36,0:36:99:0,108,1556&quot;
##      IN2009T1_us22               LBUS5                      
## [1,] &quot;1|1:0,8:8:24:324,24,0&quot;     &quot;1|1:0,6:6:18:243,18,0&quot;    
## [2,] &quot;0|0:16,0:16:48:0,48,650&quot;   &quot;0|0:20,0:20:60:0,60,819&quot;  
## [3,] &quot;0|0:23,0:23:69:0,69,946&quot;   &quot;0|0:26,0:26:78:0,78,1077&quot; 
## [4,] &quot;0|0:32,0:32:96:0,96,1299&quot;  &quot;0|0:27,0:27:81:0,81,1158&quot; 
## [5,] &quot;0|0:41,0:41:99:0,122,1613&quot; &quot;0|0:30,0:30:90:0,90,1242&quot; 
## [6,] &quot;0|0:35,0:35:99:0,105,1467&quot; &quot;0|0:36,0:36:99:0,108,1556&quot;
##      NL07434                    
## [1,] &quot;1|1:0,12:12:36:486,36,0&quot;  
## [2,] &quot;0|0:28,0:28:84:0,84,948&quot;  
## [3,] &quot;0|1:19,20:39:99:565,0,559&quot;
## [4,] &quot;0|1:19,19:38:99:523,0,535&quot;
## [5,] &quot;0|1:22,22:44:99:593,0,651&quot;
## [6,] &quot;0|1:29,25:54:99:723,0,876&quot;
## 
## *****      Var info (chromR)       *****
## *****       First 6 columns        *****
##              CHROM POS    MQ  DP  mask  n
## 1 Supercontig_1.50  41 51.30 174 FALSE NA
## 2 Supercontig_1.50 136 52.83 390  TRUE 17
## 3 Supercontig_1.50 254 56.79 514  TRUE 17
## 4 Supercontig_1.50 275 57.07 514  TRUE 17
## 5 Supercontig_1.50 386 57.40 509  TRUE 16
## 6 Supercontig_1.50 462 58.89 508  TRUE 17
## 
## *****      VCF mask (chromR)        *****
## Percent unmasked: 70.7
## 
## *****      End head (chromR)        *****</code></pre>
<p>Recall from the vignette ‘vcf data’ that the genotype information
begins at column nine from the file. This column specifies the format of
all subsequent columns. All columns subsequent to this column contain
information for each sample where each column corresponds to a sample.
We see form the output of head() that our genotype data contains
GT:AD:DP:GQ:PL data. These abbreviations are defined in the meta portion
of the file. Observation of the first few rows of genotype information
shows the format as 1|1:0,7:7:21:283,21,0. These are colon delimited
values, as in the format column. Because the genotypes are not strictly
numeric (they contain a slash to delimit unphased data and a pipe to
delimit phased data) we won’t be able to treat this as numeric
information. Read depth (DP) is numeric, and we’ll use that in
subsequent examples.</p>
</div>
<div id="using-extract.gt" class="section level2">
<h2>Using extract.gt</h2>
<p>The genotype region of VCF data contains several values for each
sample and each variant. We can extract one of these colon delimited
values using the function <strong>extract.gt()</strong>. This function
determines the position of the desired element in the FORMAT column and
uses this information to extract the same position from all of the other
samples in that row. This is performed on a row by row basis so that
files containing heterogeneous FORMAT specifications can be processed.
Here we’ve asked the function to convert the resulting matrix to a
numeric matrix. This is because we’ve determined, above, that this is
indeed numeric data. But be careful with this option! If the data are
not numeric it will do its best to convert the data and will probably
not throw an error. It’s up to the user (<em>this means you</em>) to
make sure this is a reasonable decision.</p>
<pre class="r"><code>dp &lt;- extract.gt(chrom, element=&quot;DP&quot;, as.numeric=TRUE)
rownames(dp) &lt;- 1:nrow(dp)
head(dp)</code></pre>
<pre><code>##   BL2009P4_us23 DDR7602 IN2009T1_us22 LBUS5 NL07434 P10127 P10650 P11633 P12204
## 1             7       6             8     6      12      6      4      6      1
## 2            12      20            16    20      28      9      8     11      6
## 3            27      26            23    26      39     22      8     11      6
## 4            29      27            32    27      38     22      7     10      4
## 5            26      30            41    30      44     18     11     11     NA
## 6            23      36            35    36      54     18     20     18      1
##   P13527 P1362 P13626 P17777us22 P6096 P7722 RS2009P1_us8 blue13 t30-4
## 1      7    NA     13         14     6    NA            6     16     2
## 2     29     1     41         33    11    NA           21     31     3
## 3     47     3     58         58    11    NA           28     71     2
## 4     44     5     70         62    10    NA           35     63     2
## 5     29     3     62         49    11    NA           38     60     4
## 6     37     4     48         52    18    NA           27     68     7</code></pre>
<p>Once we’ve extracted the desired element, we’ll want to visualize the
resulting matrix of data. The head() statement above gives us a peak at
what it looks like. But it doesn’t allow us to view the entire matrix.
The function <strong>heatmap.bp()</strong> helps us to visualize a
matrix of data.</p>
<pre class="r"><code>heatmap.bp(dp[1001:1500,])</code></pre>
<p><img src="visualization_2_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>The plot that results from heatmap.bp() shows samples in columns and
variants in rows. Marginal barplots summarize sums of the rows and
columns. The color ramp, at the right, is parameterized for continuous
data. In the above plot, yellow is a high number of high quality bases
violet is a low number of high quality bases. If you ask the function to
visualize more values than the screen has pixels, some form of
interpolation is performed. Because of this I suggest using this
function for no more than about 1,000 variants.</p>
</div>
<div id="interpretation-of-zeros" class="section level2">
<h2>Interpretation of zeros</h2>
<p>The presence of zeros in a dataset presents a conundrum. Sometimes,
the presence of zero valued data represents the positive determination
that there is zero valued data. For example, if we went to the grocery
store and at the checkout counter we could count that we have three
bananas, two oranges but zero apples when we know the store had apples.
This means that we know that we did not want any apples as opposed to
the store not having apples, or perhaps we just forgot to look, or a
number of other scenarios. Alternatively, the presences of zero valued
data indicates that there is missing data. This would be similar to
counting how many bananas and oranges are in our basket, but not
reporting on the number of apples. We do not know if there are zero
apples, or we just didn’t count the apples. For our sequence depth, a
zero value may indicate that the nucleotide exists in the reference
sequence but not in the resequenced sample. Alternatively, it may mean
that the resequenced sample was not sequenced at sufficient depth to
detect its presence. In the present example, we may interpret zero data
as missing data. Therefore, we should re-encode zero data as NA
data.</p>
<pre class="r"><code>is.na(dp[na.omit(dp == 0)]) &lt;- TRUE</code></pre>
<p>And re-visualize.</p>
<pre class="r"><code>heatmap.bp(dp[1001:1500,])</code></pre>
<p><img src="visualization_2_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>There are some important differences among the two above plots. In
the first plot there are missing data, but there are also sites where
there is a read depth of zero. The difference is that in the first plot,
zero data are coded as dark violet while missing data are coded as white
(a color that is not present in the color ramp). In the second plot, the
zero data were recoded as missing so they now stand out better. One of
the downsides of the heatmap is that it tells up what is high valued or
low, but it does not tell us the actual values. We can use a barplot to
inform us of this information.</p>
<pre class="r"><code>par(mar=c(8,4,4,2))
barplot(apply(dp, MARGIN=2, mean, na.rm=TRUE), las=3)</code></pre>
<p><img src="visualization_2_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<pre class="r"><code>par(mar=c(5,4,4,2))</code></pre>
<p>Here we see that the mean number of high quality bases per variant is
quite high for most samples. The sample p1362 appears to have an
abundance of missing data in the heatmap. In the barplot we’ve validated
that this sample lacks information. At the other end of the spectrum,
the sample NL07434 has a high average coverage as seen in the heatmap,
and in the barplot we can see that this value is just over 40.</p>
</div>
<div id="conclusion" class="section level2">
<h2>Conclusion</h2>
<p>Now that we know our way around our chromosome it may be time to make
some decisions. Regions which do not meet our expectations may need
extra work to make sure they are of high quality. Alternatively, we may
simply wish to censor these regions and focus on the high quality
fraction of the data. Once we’ve gained a familiarity with the data
contained in our VCF file we can make informed decisions on how best to
process it.</p>
</div>

<center>
<hr class="style1">
<p>Copyright &copy; 2017, 2018 Brian J. Knaus. All rights reserved.</p>
<p>USDA Agricultural Research Service, Horticultural Crops Research Lab.</p>
</center>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
