<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Brian J. Knaus" />

<meta name="date" content="2017-09-27" />

<title>Windowing</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<!— Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-1”></script>
<script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments)}; gtag('js', new Date()); gtag('config', 'UA-107144798-1'); </script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">vcfR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="rlanguage.html">R language</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="quick_intro.html">A quick introduction</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="vcfR_object.html">vcfR objects</a>
    </li>
    <li>
      <a href="how_much_memory.html">How much memory</a>
    </li>
    <li>
      <a href="matrices.html">Extracting matrices</a>
    </li>
    <li>
      <a href="tidy_vcfR.html">Tidy vcfR</a>
    </li>
    <li>
      <a href="chromR_object.html">chromR objects</a>
    </li>
    <li>
      <a href="visualization_1.html">Visualization 1</a>
    </li>
    <li>
      <a href="visualization_2.html">Visualization 2</a>
    </li>
    <li>
      <a href="sequence_coverage.html">Sequence coverage</a>
    </li>
    <li>
      <a href="filtering_data.html">Filtering data</a>
    </li>
    <li>
      <a href="ranking_data.html">Ranking data</a>
    </li>
    <li>
      <a href="windowing.html">Windowing</a>
    </li>
    <li>
      <a href="genetic_differentiation.html">Genetic differentiation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    GBS class
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="gbs_class.html">GBS class</a>
    </li>
    <li>
      <a href="vcf_data.html">VCF data</a>
    </li>
    <li>
      <a href="extract_data.html">Extract data</a>
    </li>
    <li>
      <a href="depth_plot.html">Depth plot</a>
    </li>
    <li>
      <a href="missing_data.html">Missing data</a>
    </li>
    <li>
      <a href="censoring_data.html">Censoring data</a>
    </li>
    <li>
      <a href="omitting_data.html">Omitting data</a>
    </li>
    <li>
      <a href="apply.html">Apply</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Ploidy
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ploidy.html">Ploidy</a>
    </li>
    <li>
      <a href="determining_ploidy_1.html">Determining ploidy 1</a>
    </li>
    <li>
      <a href="determining_ploidy_2.html">Determining ploidy 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Export
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="export.html">Overview</a>
    </li>
    <li>
      <a href="export_vcfgz.html">Export to *vcf.gz</a>
    </li>
    <li>
      <a href="export_genind_genclone.html">Genind and Genclone</a>
    </li>
    <li>
      <a href="export_genlight_snpclone.html">Genlight and SNPclone</a>
    </li>
    <li>
      <a href="dnabin.html">DNAbin</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="subset_data_to_1chrom.html">Subset to 1 chromosome</a>
    </li>
    <li>
      <a href="vcf_software.html">VCF software</a>
    </li>
    <li>
      <a href="dip_to_hap.html">Haploidizing diploid data</a>
    </li>
    <li>
      <a href="compiling_vcfR.html">Compiling vcfR</a>
    </li>
    <li>
      <a href="reporting_issue.html">Reporting an issue</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<p>
<center>
<h3>vcfR documentation</h3>
by
<br>
Brian J. Knaus and Niklaus J. Gr&uuml;nwald
</center>
</p>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Windowing</h1>
<h4 class="author"><em>Brian J. Knaus</em></h4>
<h4 class="date"><em>2017-09-27</em></h4>

</div>


<p>The package vcfR was designed to work on individual chromosomes. This is because a chromosome is a convenient way to subset a genome into smaller tasks. We are frequently interested in examining an entire genome in order to identify features of interest. This can be accomplished by processing chromosomes one at a time and writing summaries to a file. This allows us to survey entire genomes even when we lack the computational resources necessary to load entire genomes into memory.</p>
<div id="example-data" class="section level2">
<h2>Example data</h2>
<p>vcfR comes with several example datasets. Here we’ll use them to create a small example genomic dataset.</p>
<pre class="r"><code>library(vcfR)
data(vcfR_example)
data(&quot;vcfR_test&quot;)</code></pre>
<p>The vcfR_test dataset consists only of VCF data. We’ll fabricate a sequence and annotation for the sake of an example.</p>
<pre class="r"><code>library(ape)

dna.l &lt;- vector(&#39;list&#39;, length=3)

dna.l[[1]] &lt;- as.character(dna[1,1:length(dna)])
set.seed(123)
dna.l[[2]] &lt;- sample( c(&#39;a&#39;,&#39;c&#39;,&#39;g&#39;,&#39;t&#39;), size = getPOS(vcfR_test)[length(getPOS(vcfR_test))], replace = TRUE )
dna.l[[3]] &lt;- sample( c(&#39;a&#39;,&#39;c&#39;,&#39;g&#39;,&#39;t&#39;), size = getPOS(vcfR_test)[length(getPOS(vcfR_test))], replace = TRUE )
dna.l &lt;- as.DNAbin(dna.l)

names(dna.l)[1] &lt;- &quot;Supercontig_1.50&quot;
names(dna.l)[2] &lt;- &quot;Supercontig_1.5&quot;
names(dna.l)[3] &lt;- &quot;Supercontig_1.10&quot;
dna.l</code></pre>
<pre><code>## 3 DNA sequences in binary format stored in a list.
## 
## Mean sequence length: 856378.3 
##    Shortest sequence: 100001 
##     Longest sequence: 1234567 
## 
## Labels:
## Supercontig_1.50
## Supercontig_1.5
## Supercontig_1.10
## 
## Base composition:
##    a    c    g    t 
## 0.25 0.25 0.25 0.25</code></pre>
<pre class="r"><code>gff2 &lt;- gff[1:10,]
gff2[,1] &lt;- &quot;Supercontig_1.5&quot;
gff3 &lt;- gff[11:20,]
gff3[,1] &lt;- &quot;Supercontig_1.10&quot;</code></pre>
<p>I’m used to receiving reference and annotation data as a single file each. We’ll combine those data here for the example.</p>
<pre class="r"><code>gff &lt;- rbind(gff, gff2, gff3)
rm(gff2)
rm(gff3)</code></pre>
<p>I’m also used to having my VCF data seperated as one file per chromosome. So we’ll write our VCF data to file.</p>
<pre class="r"><code>write.vcf(vcf, file = &quot;Supercontig_1.50.vcf.gz&quot;)
vcfR_test@fix[,&#39;CHROM&#39;] &lt;- &quot;Supercontig_1.5&quot;
write.vcf(vcfR_test, file = &quot;Supercontig_1.5.vcf.gz&quot;)
vcfR_test@fix[,&#39;CHROM&#39;] &lt;- &quot;Supercontig_1.10&quot;
write.vcf(vcfR_test, file = &quot;Supercontig_1.10.vcf.gz&quot;)</code></pre>
</div>
<div id="windowing-through-chromosomes" class="section level2">
<h2>Windowing through chromosomes</h2>
<p>We now have a reference sequence and annotation information in memory and VCF data in files. We are going to use vcfR to windowize each chromosome and write summaries for each window and each variant to file. This is accomplished by first writing the files to disk and subsequently appending to each file.</p>
<pre class="r"><code>myFiles &lt;- list.files(&quot;.&quot;, pattern = &quot;Supercontig.*vcf.gz$&quot;)
myChroms &lt;- unlist( lapply( strsplit(myFiles, &quot;\\.vcf&quot;), function(x){ x[1] } ) )

myWin.size &lt;- 1e4
i &lt;- 1
myChrom &lt;- myChroms[i]

dna1 &lt;- dna.l[myChrom]
gff1 &lt;- gff[ gff[,1] == myChrom,]

vcf &lt;- read.vcfR(myFiles[i], verbose = FALSE)
    
chrom &lt;- create.chromR(name=myChrom, vcf=vcf, seq=dna1, ann=gff1, verbose = FALSE)
chrom &lt;- proc.chromR(chrom, win.size = myWin.size, verbose=FALSE)

write.var.info(chrom, file = &quot;pinf_ref_vars.csv&quot;, APPEND = FALSE)
write.win.info(chrom, file = &quot;pinf_ref_wins.csv&quot;, APPEND = FALSE)

for(i in 2:length(myFiles)){
  myChrom &lt;- myChroms[i]
  cat(myChrom)
  cat(&quot;\n&quot;)
  
  dna1 &lt;- dna.l[myChrom]
  gff1 &lt;- gff[ gff[,1] == myChrom,]

  vcf &lt;- read.vcfR(myFiles[i], verbose = FALSE)

  chrom &lt;- create.chromR(name=myChrom, vcf=vcf, seq=dna1, ann=gff1, verbose = FALSE)
  chrom &lt;- proc.chromR(chrom, win.size = myWin.size, verbose=FALSE)

  write.var.info(chrom, file = &quot;pinf_ref_vars.csv&quot;, APPEND = TRUE)
  write.win.info(chrom, file = &quot;pinf_ref_wins.csv&quot;, APPEND = TRUE)

}</code></pre>
<pre><code>## Supercontig_1.50
## Supercontig_1.5</code></pre>
</div>
<div id="clean-up" class="section level2">
<h2>Clean up</h2>
<p>We wrote two files to disk for this example. Typically these files would be data and would therefore be valuable. In the context of this example, we no longer need them, so we’ll delete them. Make sure you do not delete your real data!</p>
<pre class="r"><code>unlink(&quot;Supercontig_1.5.vcf.gz&quot;)
unlink(&quot;Supercontig_1.10.vcf.gz&quot;)
unlink(&quot;Supercontig_1.50.vcf.gz&quot;)

unlink(&quot;pinf_ref_vars.csv&quot;)
unlink(&quot;pinf_ref_wins.csv&quot;)</code></pre>
</div>

<center>
<hr class="style1">
<p>Copyright &copy; 2017 Brian J. Knaus. All rights reserved.</p>
<p>USDA Agricultural Research Service, Horticultural Crops Research Lab.</p>
</center>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
